// create-vue/bundle.js
#!/usr/bin/env node
/*! create-vue v3.21.1 | MIT */
import { createRequire } from "node:module";
import * as fs from "node:fs";
import * as path$1 from "node:path";
import { fileURLToPath, pathToFileURL } from "node:url";
import { parseArgs, stripVTControlCharacters } from "node:util";
import y, { stdin, stdout } from "node:process";
import * as g from "node:readline";
import O from "node:readline";
import { Writable } from "node:stream";
import * as path from "path";

//#region \0rolldown/runtime.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJSMin = (cb, mod) => () => (mod || cb((mod = { exports: {} }).exports, mod), mod.exports);
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") {
		for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
			key = keys[i];
			if (!__hasOwnProp.call(to, key) && key !== except) {
				__defProp(to, key, {
					get: ((k) => from[k]).bind(null, key),
					enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
				});
			}
		}
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));
var __require = /* @__PURE__ */ createRequire(import.meta.url);

//#endregion
//#region node_modules/.pnpm/sisteransi@1.0.5/node_modules/sisteransi/src/index.js
var require_src = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	const ESC = "\x1B";
	const CSI = `${ESC}[`;
	const beep = "\x07";
	const cursor = {
		to(x, y) {
			if (!y) return `${CSI}${x + 1}G`;
			return `${CSI}${y + 1};${x + 1}H`;
		},
		move(x, y) {
			let ret = "";
			if (x < 0) ret += `${CSI}${-x}D`;
			else if (x > 0) ret += `${CSI}${x}C`;
			if (y < 0) ret += `${CSI}${-y}A`;
			else if (y > 0) ret += `${CSI}${y}B`;
			return ret;
		},
		up: (count = 1) => `${CSI}${count}A`,
		down: (count = 1) => `${CSI}${count}B`,
		forward: (count = 1) => `${CSI}${count}C`,
		backward: (count = 1) => `${CSI}${count}D`,
		nextLine: (count = 1) => `${CSI}E`.repeat(count),
		prevLine: (count = 1) => `${CSI}F`.repeat(count),
		left: `${CSI}G`,
		hide: `${CSI}?25l`,
		show: `${CSI}?25h`,
		save: `${ESC}7`,
		restore: `${ESC}8`
	};
	const scroll = {
		up: (count = 1) => `${CSI}S`.repeat(count),
		down: (count = 1) => `${CSI}T`.repeat(count)
	};
	const erase = {
		screen: `${CSI}2J`,
		up: (count = 1) => `${CSI}1J`.repeat(count),
		down: (count = 1) => `${CSI}J`.repeat(count),
		line: `${CSI}2K`,
		lineEnd: `${CSI}K`,
		lineStart: `${CSI}1K`,
		lines(count) {
			let clear = "";
			for (let i = 0; i < count; i++) clear += this.line + (i < count - 1 ? cursor.up() : "");
			if (count) clear += cursor.left;
			return clear;
		}
	};
	module.exports = {
		cursor,
		scroll,
		erase,
		beep
	};
}));

//#endregion
//#region node_modules/.pnpm/picocolors@1.1.1/node_modules/picocolors/picocolors.js
var require_picocolors = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	let p = process || {}, argv = p.argv || [], env = p.env || {};
	let isColorSupported = !(!!env.NO_COLOR || argv.includes("--no-color")) && (!!env.FORCE_COLOR || argv.includes("--color") || p.platform === "win32" || (p.stdout || {}).isTTY && env.TERM !== "dumb" || !!env.CI);
	let formatter = (open, close, replace = open) => (input) => {
		let string = "" + input, index = string.indexOf(close, open.length);
		return ~index ? open + replaceClose(string, close, replace, index) + close : open + string + close;
	};
	let replaceClose = (string, close, replace, index) => {
		let result = "", cursor = 0;
		do {
			result += string.substring(cursor, index) + replace;
			cursor = index + close.length;
			index = string.indexOf(close, cursor);
		} while (~index);
		return result + string.substring(cursor);
	};
	let createColors = (enabled = isColorSupported) => {
		let f = enabled ? formatter : () => String;
		return {
			isColorSupported: enabled,
			reset: f("\x1B[0m", "\x1B[0m"),
			bold: f("\x1B[1m", "\x1B[22m", "\x1B[22m\x1B[1m"),
			dim: f("\x1B[2m", "\x1B[22m", "\x1B[22m\x1B[2m"),
			italic: f("\x1B[3m", "\x1B[23m"),
			underline: f("\x1B[4m", "\x1B[24m"),
			inverse: f("\x1B[7m", "\x1B[27m"),
			hidden: f("\x1B[8m", "\x1B[28m"),
			strikethrough: f("\x1B[9m", "\x1B[29m"),
			black: f("\x1B[30m", "\x1B[39m"),
			red: f("\x1B[31m", "\x1B[39m"),
			green: f("\x1B[32m", "\x1B[39m"),
			yellow: f("\x1B[33m", "\x1B[39m"),
			blue: f("\x1B[34m", "\x1B[39m"),
			magenta: f("\x1B[35m", "\x1B[39m"),
			cyan: f("\x1B[36m", "\x1B[39m"),
			white: f("\x1B[37m", "\x1B[39m"),
			gray: f("\x1B[90m", "\x1B[39m"),
			bgBlack: f("\x1B[40m", "\x1B[49m"),
			bgRed: f("\x1B[41m", "\x1B[49m"),
			bgGreen: f("\x1B[42m", "\x1B[49m"),
			bgYellow: f("\x1B[43m", "\x1B[49m"),
			bgBlue: f("\x1B[44m", "\x1B[49m"),
			bgMagenta: f("\x1B[45m", "\x1B[49m"),
			bgCyan: f("\x1B[46m", "\x1B[49m"),
			bgWhite: f("\x1B[47m", "\x1B[49m"),
			blackBright: f("\x1B[90m", "\x1B[39m"),
			redBright: f("\x1B[91m", "\x1B[39m"),
			greenBright: f("\x1B[92m", "\x1B[39m"),
			yellowBright: f("\x1B[93m", "\x1B[39m"),
			blueBright: f("\x1B[94m", "\x1B[39m"),
			magentaBright: f("\x1B[95m", "\x1B[39m"),
			cyanBright: f("\x1B[96m", "\x1B[39m"),
			whiteBright: f("\x1B[97m", "\x1B[39m"),
			bgBlackBright: f("\x1B[100m", "\x1B[49m"),
			bgRedBright: f("\x1B[101m", "\x1B[49m"),
			bgGreenBright: f("\x1B[102m", "\x1B[49m"),
			bgYellowBright: f("\x1B[103m", "\x1B[49m"),
			bgBlueBright: f("\x1B[104m", "\x1B[49m"),
			bgMagentaBright: f("\x1B[105m", "\x1B[49m"),
			bgCyanBright: f("\x1B[106m", "\x1B[49m"),
			bgWhiteBright: f("\x1B[107m", "\x1B[49m")
		};
	};
	module.exports = createColors();
	module.exports.createColors = createColors;
}));

//#endregion
//#region node_modules/.pnpm/@clack+core@0.5.0/node_modules/@clack/core/dist/index.mjs
var import_src = require_src();
var import_picocolors = /* @__PURE__ */ __toESM(require_picocolors(), 1);
function DD({ onlyFirst: e = !1 } = {}) {
	const t = ["[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]+)*|[a-zA-Z\\d]+(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?(?:\\u0007|\\u001B\\u005C|\\u009C))", "(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-nq-uy=><~]))"].join("|");
	return new RegExp(t, e ? void 0 : "g");
}
const uD = DD();
function P$1(e) {
	if (typeof e != "string") throw new TypeError(`Expected a \`string\`, got \`${typeof e}\``);
	return e.replace(uD, "");
}
function L$1(e) {
	return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var W$1 = { exports: {} };
(function(e) {
	var u = {};
	e.exports = u, u.eastAsianWidth = function(F) {
		var s = F.charCodeAt(0), i = F.length == 2 ? F.charCodeAt(1) : 0, D = s;
		return 55296 <= s && s <= 56319 && 56320 <= i && i <= 57343 && (s &= 1023, i &= 1023, D = s << 10 | i, D += 65536), D == 12288 || 65281 <= D && D <= 65376 || 65504 <= D && D <= 65510 ? "F" : D == 8361 || 65377 <= D && D <= 65470 || 65474 <= D && D <= 65479 || 65482 <= D && D <= 65487 || 65490 <= D && D <= 65495 || 65498 <= D && D <= 65500 || 65512 <= D && D <= 65518 ? "H" : 4352 <= D && D <= 4447 || 4515 <= D && D <= 4519 || 4602 <= D && D <= 4607 || 9001 <= D && D <= 9002 || 11904 <= D && D <= 11929 || 11931 <= D && D <= 12019 || 12032 <= D && D <= 12245 || 12272 <= D && D <= 12283 || 12289 <= D && D <= 12350 || 12353 <= D && D <= 12438 || 12441 <= D && D <= 12543 || 12549 <= D && D <= 12589 || 12593 <= D && D <= 12686 || 12688 <= D && D <= 12730 || 12736 <= D && D <= 12771 || 12784 <= D && D <= 12830 || 12832 <= D && D <= 12871 || 12880 <= D && D <= 13054 || 13056 <= D && D <= 19903 || 19968 <= D && D <= 42124 || 42128 <= D && D <= 42182 || 43360 <= D && D <= 43388 || 44032 <= D && D <= 55203 || 55216 <= D && D <= 55238 || 55243 <= D && D <= 55291 || 63744 <= D && D <= 64255 || 65040 <= D && D <= 65049 || 65072 <= D && D <= 65106 || 65108 <= D && D <= 65126 || 65128 <= D && D <= 65131 || 110592 <= D && D <= 110593 || 127488 <= D && D <= 127490 || 127504 <= D && D <= 127546 || 127552 <= D && D <= 127560 || 127568 <= D && D <= 127569 || 131072 <= D && D <= 194367 || 177984 <= D && D <= 196605 || 196608 <= D && D <= 262141 ? "W" : 32 <= D && D <= 126 || 162 <= D && D <= 163 || 165 <= D && D <= 166 || D == 172 || D == 175 || 10214 <= D && D <= 10221 || 10629 <= D && D <= 10630 ? "Na" : D == 161 || D == 164 || 167 <= D && D <= 168 || D == 170 || 173 <= D && D <= 174 || 176 <= D && D <= 180 || 182 <= D && D <= 186 || 188 <= D && D <= 191 || D == 198 || D == 208 || 215 <= D && D <= 216 || 222 <= D && D <= 225 || D == 230 || 232 <= D && D <= 234 || 236 <= D && D <= 237 || D == 240 || 242 <= D && D <= 243 || 247 <= D && D <= 250 || D == 252 || D == 254 || D == 257 || D == 273 || D == 275 || D == 283 || 294 <= D && D <= 295 || D == 299 || 305 <= D && D <= 307 || D == 312 || 319 <= D && D <= 322 || D == 324 || 328 <= D && D <= 331 || D == 333 || 338 <= D && D <= 339 || 358 <= D && D <= 359 || D == 363 || D == 462 || D == 464 || D == 466 || D == 468 || D == 470 || D == 472 || D == 474 || D == 476 || D == 593 || D == 609 || D == 708 || D == 711 || 713 <= D && D <= 715 || D == 717 || D == 720 || 728 <= D && D <= 731 || D == 733 || D == 735 || 768 <= D && D <= 879 || 913 <= D && D <= 929 || 931 <= D && D <= 937 || 945 <= D && D <= 961 || 963 <= D && D <= 969 || D == 1025 || 1040 <= D && D <= 1103 || D == 1105 || D == 8208 || 8211 <= D && D <= 8214 || 8216 <= D && D <= 8217 || 8220 <= D && D <= 8221 || 8224 <= D && D <= 8226 || 8228 <= D && D <= 8231 || D == 8240 || 8242 <= D && D <= 8243 || D == 8245 || D == 8251 || D == 8254 || D == 8308 || D == 8319 || 8321 <= D && D <= 8324 || D == 8364 || D == 8451 || D == 8453 || D == 8457 || D == 8467 || D == 8470 || 8481 <= D && D <= 8482 || D == 8486 || D == 8491 || 8531 <= D && D <= 8532 || 8539 <= D && D <= 8542 || 8544 <= D && D <= 8555 || 8560 <= D && D <= 8569 || D == 8585 || 8592 <= D && D <= 8601 || 8632 <= D && D <= 8633 || D == 8658 || D == 8660 || D == 8679 || D == 8704 || 8706 <= D && D <= 8707 || 8711 <= D && D <= 8712 || D == 8715 || D == 8719 || D == 8721 || D == 8725 || D == 8730 || 8733 <= D && D <= 8736 || D == 8739 || D == 8741 || 8743 <= D && D <= 8748 || D == 8750 || 8756 <= D && D <= 8759 || 8764 <= D && D <= 8765 || D == 8776 || D == 8780 || D == 8786 || 8800 <= D && D <= 8801 || 8804 <= D && D <= 8807 || 8810 <= D && D <= 8811 || 8814 <= D && D <= 8815 || 8834 <= D && D <= 8835 || 8838 <= D && D <= 8839 || D == 8853 || D == 8857 || D == 8869 || D == 8895 || D == 8978 || 9312 <= D && D <= 9449 || 9451 <= D && D <= 9547 || 9552 <= D && D <= 9587 || 9600 <= D && D <= 9615 || 9618 <= D && D <= 9621 || 9632 <= D && D <= 9633 || 9635 <= D && D <= 9641 || 9650 <= D && D <= 9651 || 9654 <= D && D <= 9655 || 9660 <= D && D <= 9661 || 9664 <= D && D <= 9665 || 9670 <= D && D <= 9672 || D == 9675 || 9678 <= D && D <= 9681 || 9698 <= D && D <= 9701 || D == 9711 || 9733 <= D && D <= 9734 || D == 9737 || 9742 <= D && D <= 9743 || 9748 <= D && D <= 9749 || D == 9756 || D == 9758 || D == 9792 || D == 9794 || 9824 <= D && D <= 9825 || 9827 <= D && D <= 9829 || 9831 <= D && D <= 9834 || 9836 <= D && D <= 9837 || D == 9839 || 9886 <= D && D <= 9887 || 9918 <= D && D <= 9919 || 9924 <= D && D <= 9933 || 9935 <= D && D <= 9953 || D == 9955 || 9960 <= D && D <= 9983 || D == 10045 || D == 10071 || 10102 <= D && D <= 10111 || 11093 <= D && D <= 11097 || 12872 <= D && D <= 12879 || 57344 <= D && D <= 63743 || 65024 <= D && D <= 65039 || D == 65533 || 127232 <= D && D <= 127242 || 127248 <= D && D <= 127277 || 127280 <= D && D <= 127337 || 127344 <= D && D <= 127386 || 917760 <= D && D <= 917999 || 983040 <= D && D <= 1048573 || 1048576 <= D && D <= 1114109 ? "A" : "N";
	}, u.characterLength = function(F) {
		var s = this.eastAsianWidth(F);
		return s == "F" || s == "W" || s == "A" ? 2 : 1;
	};
	function t(F) {
		return F.match(/[\uD800-\uDBFF][\uDC00-\uDFFF]|[^\uD800-\uDFFF]/g) || [];
	}
	u.length = function(F) {
		for (var s = t(F), i = 0, D = 0; D < s.length; D++) i = i + this.characterLength(s[D]);
		return i;
	}, u.slice = function(F, s, i) {
		textLen = u.length(F), s = s || 0, i = i || 1, s < 0 && (s = textLen + s), i < 0 && (i = textLen + i);
		for (var D = "", C = 0, n = t(F), E = 0; E < n.length; E++) {
			var a = n[E], o = u.length(a);
			if (C >= s - (o == 2 ? 1 : 0)) if (C + o <= i) D += a;
			else break;
			C += o;
		}
		return D;
	};
})(W$1);
var tD = W$1.exports;
const eD = L$1(tD);
var FD = function() {
	return /\uD83C\uDFF4\uDB40\uDC67\uDB40\uDC62(?:\uDB40\uDC77\uDB40\uDC6C\uDB40\uDC73|\uDB40\uDC73\uDB40\uDC63\uDB40\uDC74|\uDB40\uDC65\uDB40\uDC6E\uDB40\uDC67)\uDB40\uDC7F|(?:\uD83E\uDDD1\uD83C\uDFFF\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFF\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB-\uDFFE])|(?:\uD83E\uDDD1\uD83C\uDFFE\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFE\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB-\uDFFD\uDFFF])|(?:\uD83E\uDDD1\uD83C\uDFFD\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFD\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB\uDFFC\uDFFE\uDFFF])|(?:\uD83E\uDDD1\uD83C\uDFFC\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFC\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB\uDFFD-\uDFFF])|(?:\uD83E\uDDD1\uD83C\uDFFB\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFB\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFC-\uDFFF])|\uD83D\uDC68(?:\uD83C\uDFFB(?:\u200D(?:\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF]))|\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFC-\uDFFF])|[\u2695\u2696\u2708]\uFE0F|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD]))?|(?:\uD83C[\uDFFC-\uDFFF])\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF]))|\u200D(?:\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83D\uDC68|(?:\uD83D[\uDC68\uDC69])\u200D(?:\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67]))|\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFF\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFE])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFE\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFD\uDFFF])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFD\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB\uDFFC\uDFFE\uDFFF])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFC\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB\uDFFD-\uDFFF])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|(?:\uD83C\uDFFF\u200D[\u2695\u2696\u2708]|\uD83C\uDFFE\u200D[\u2695\u2696\u2708]|\uD83C\uDFFD\u200D[\u2695\u2696\u2708]|\uD83C\uDFFC\u200D[\u2695\u2696\u2708]|\u200D[\u2695\u2696\u2708])\uFE0F|\u200D(?:(?:\uD83D[\uDC68\uDC69])\u200D(?:\uD83D[\uDC66\uDC67])|\uD83D[\uDC66\uDC67])|\uD83C\uDFFF|\uD83C\uDFFE|\uD83C\uDFFD|\uD83C\uDFFC)?|(?:\uD83D\uDC69(?:\uD83C\uDFFB\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D(?:\uD83D[\uDC68\uDC69])|\uD83D[\uDC68\uDC69])|(?:\uD83C[\uDFFC-\uDFFF])\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D(?:\uD83D[\uDC68\uDC69])|\uD83D[\uDC68\uDC69]))|\uD83E\uDDD1(?:\uD83C[\uDFFB-\uDFFF])\u200D\uD83E\uDD1D\u200D\uD83E\uDDD1)(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC69\u200D\uD83D\uDC69\u200D(?:\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67]))|\uD83D\uDC69(?:\u200D(?:\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D(?:\uD83D[\uDC68\uDC69])|\uD83D[\uDC68\uDC69])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFF\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFE\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFD\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFC\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFB\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD]))|\uD83E\uDDD1(?:\u200D(?:\uD83E\uDD1D\u200D\uD83E\uDDD1|\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFF\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFE\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFD\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFC\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFB\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD]))|\uD83D\uDC69\u200D\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC69\u200D\uD83D\uDC69\u200D(?:\uD83D[\uDC66\uDC67])|\uD83D\uDC69\u200D\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67])|(?:\uD83D\uDC41\uFE0F\u200D\uD83D\uDDE8|\uD83E\uDDD1(?:\uD83C\uDFFF\u200D[\u2695\u2696\u2708]|\uD83C\uDFFE\u200D[\u2695\u2696\u2708]|\uD83C\uDFFD\u200D[\u2695\u2696\u2708]|\uD83C\uDFFC\u200D[\u2695\u2696\u2708]|\uD83C\uDFFB\u200D[\u2695\u2696\u2708]|\u200D[\u2695\u2696\u2708])|\uD83D\uDC69(?:\uD83C\uDFFF\u200D[\u2695\u2696\u2708]|\uD83C\uDFFE\u200D[\u2695\u2696\u2708]|\uD83C\uDFFD\u200D[\u2695\u2696\u2708]|\uD83C\uDFFC\u200D[\u2695\u2696\u2708]|\uD83C\uDFFB\u200D[\u2695\u2696\u2708]|\u200D[\u2695\u2696\u2708])|\uD83D\uDE36\u200D\uD83C\uDF2B|\uD83C\uDFF3\uFE0F\u200D\u26A7|\uD83D\uDC3B\u200D\u2744|(?:(?:\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD])(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC6F|\uD83E[\uDD3C\uDDDE\uDDDF])\u200D[\u2640\u2642]|(?:\u26F9|\uD83C[\uDFCB\uDFCC]|\uD83D\uDD75)(?:\uFE0F|\uD83C[\uDFFB-\uDFFF])\u200D[\u2640\u2642]|\uD83C\uDFF4\u200D\u2620|(?:\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD])\u200D[\u2640\u2642]|[\xA9\xAE\u203C\u2049\u2122\u2139\u2194-\u2199\u21A9\u21AA\u2328\u23CF\u23ED-\u23EF\u23F1\u23F2\u23F8-\u23FA\u24C2\u25AA\u25AB\u25B6\u25C0\u25FB\u25FC\u2600-\u2604\u260E\u2611\u2618\u2620\u2622\u2623\u2626\u262A\u262E\u262F\u2638-\u263A\u2640\u2642\u265F\u2660\u2663\u2665\u2666\u2668\u267B\u267E\u2692\u2694-\u2697\u2699\u269B\u269C\u26A0\u26A7\u26B0\u26B1\u26C8\u26CF\u26D1\u26D3\u26E9\u26F0\u26F1\u26F4\u26F7\u26F8\u2702\u2708\u2709\u270F\u2712\u2714\u2716\u271D\u2721\u2733\u2734\u2744\u2747\u2763\u27A1\u2934\u2935\u2B05-\u2B07\u3030\u303D\u3297\u3299]|\uD83C[\uDD70\uDD71\uDD7E\uDD7F\uDE02\uDE37\uDF21\uDF24-\uDF2C\uDF36\uDF7D\uDF96\uDF97\uDF99-\uDF9B\uDF9E\uDF9F\uDFCD\uDFCE\uDFD4-\uDFDF\uDFF5\uDFF7]|\uD83D[\uDC3F\uDCFD\uDD49\uDD4A\uDD6F\uDD70\uDD73\uDD76-\uDD79\uDD87\uDD8A-\uDD8D\uDDA5\uDDA8\uDDB1\uDDB2\uDDBC\uDDC2-\uDDC4\uDDD1-\uDDD3\uDDDC-\uDDDE\uDDE1\uDDE3\uDDE8\uDDEF\uDDF3\uDDFA\uDECB\uDECD-\uDECF\uDEE0-\uDEE5\uDEE9\uDEF0\uDEF3])\uFE0F|\uD83C\uDFF3\uFE0F\u200D\uD83C\uDF08|\uD83D\uDC69\u200D\uD83D\uDC67|\uD83D\uDC69\u200D\uD83D\uDC66|\uD83D\uDE35\u200D\uD83D\uDCAB|\uD83D\uDE2E\u200D\uD83D\uDCA8|\uD83D\uDC15\u200D\uD83E\uDDBA|\uD83E\uDDD1(?:\uD83C\uDFFF|\uD83C\uDFFE|\uD83C\uDFFD|\uD83C\uDFFC|\uD83C\uDFFB)?|\uD83D\uDC69(?:\uD83C\uDFFF|\uD83C\uDFFE|\uD83C\uDFFD|\uD83C\uDFFC|\uD83C\uDFFB)?|\uD83C\uDDFD\uD83C\uDDF0|\uD83C\uDDF6\uD83C\uDDE6|\uD83C\uDDF4\uD83C\uDDF2|\uD83D\uDC08\u200D\u2B1B|\u2764\uFE0F\u200D(?:\uD83D\uDD25|\uD83E\uDE79)|\uD83D\uDC41\uFE0F|\uD83C\uDFF3\uFE0F|\uD83C\uDDFF(?:\uD83C[\uDDE6\uDDF2\uDDFC])|\uD83C\uDDFE(?:\uD83C[\uDDEA\uDDF9])|\uD83C\uDDFC(?:\uD83C[\uDDEB\uDDF8])|\uD83C\uDDFB(?:\uD83C[\uDDE6\uDDE8\uDDEA\uDDEC\uDDEE\uDDF3\uDDFA])|\uD83C\uDDFA(?:\uD83C[\uDDE6\uDDEC\uDDF2\uDDF3\uDDF8\uDDFE\uDDFF])|\uD83C\uDDF9(?:\uD83C[\uDDE6\uDDE8\uDDE9\uDDEB-\uDDED\uDDEF-\uDDF4\uDDF7\uDDF9\uDDFB\uDDFC\uDDFF])|\uD83C\uDDF8(?:\uD83C[\uDDE6-\uDDEA\uDDEC-\uDDF4\uDDF7-\uDDF9\uDDFB\uDDFD-\uDDFF])|\uD83C\uDDF7(?:\uD83C[\uDDEA\uDDF4\uDDF8\uDDFA\uDDFC])|\uD83C\uDDF5(?:\uD83C[\uDDE6\uDDEA-\uDDED\uDDF0-\uDDF3\uDDF7-\uDDF9\uDDFC\uDDFE])|\uD83C\uDDF3(?:\uD83C[\uDDE6\uDDE8\uDDEA-\uDDEC\uDDEE\uDDF1\uDDF4\uDDF5\uDDF7\uDDFA\uDDFF])|\uD83C\uDDF2(?:\uD83C[\uDDE6\uDDE8-\uDDED\uDDF0-\uDDFF])|\uD83C\uDDF1(?:\uD83C[\uDDE6-\uDDE8\uDDEE\uDDF0\uDDF7-\uDDFB\uDDFE])|\uD83C\uDDF0(?:\uD83C[\uDDEA\uDDEC-\uDDEE\uDDF2\uDDF3\uDDF5\uDDF7\uDDFC\uDDFE\uDDFF])|\uD83C\uDDEF(?:\uD83C[\uDDEA\uDDF2\uDDF4\uDDF5])|\uD83C\uDDEE(?:\uD83C[\uDDE8-\uDDEA\uDDF1-\uDDF4\uDDF6-\uDDF9])|\uD83C\uDDED(?:\uD83C[\uDDF0\uDDF2\uDDF3\uDDF7\uDDF9\uDDFA])|\uD83C\uDDEC(?:\uD83C[\uDDE6\uDDE7\uDDE9-\uDDEE\uDDF1-\uDDF3\uDDF5-\uDDFA\uDDFC\uDDFE])|\uD83C\uDDEB(?:\uD83C[\uDDEE-\uDDF0\uDDF2\uDDF4\uDDF7])|\uD83C\uDDEA(?:\uD83C[\uDDE6\uDDE8\uDDEA\uDDEC\uDDED\uDDF7-\uDDFA])|\uD83C\uDDE9(?:\uD83C[\uDDEA\uDDEC\uDDEF\uDDF0\uDDF2\uDDF4\uDDFF])|\uD83C\uDDE8(?:\uD83C[\uDDE6\uDDE8\uDDE9\uDDEB-\uDDEE\uDDF0-\uDDF5\uDDF7\uDDFA-\uDDFF])|\uD83C\uDDE7(?:\uD83C[\uDDE6\uDDE7\uDDE9-\uDDEF\uDDF1-\uDDF4\uDDF6-\uDDF9\uDDFB\uDDFC\uDDFE\uDDFF])|\uD83C\uDDE6(?:\uD83C[\uDDE8-\uDDEC\uDDEE\uDDF1\uDDF2\uDDF4\uDDF6-\uDDFA\uDDFC\uDDFD\uDDFF])|[#\*0-9]\uFE0F\u20E3|\u2764\uFE0F|(?:\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD])(?:\uD83C[\uDFFB-\uDFFF])|(?:\u26F9|\uD83C[\uDFCB\uDFCC]|\uD83D\uDD75)(?:\uFE0F|\uD83C[\uDFFB-\uDFFF])|\uD83C\uDFF4|(?:[\u270A\u270B]|\uD83C[\uDF85\uDFC2\uDFC7]|\uD83D[\uDC42\uDC43\uDC46-\uDC50\uDC66\uDC67\uDC6B-\uDC6D\uDC72\uDC74-\uDC76\uDC78\uDC7C\uDC83\uDC85\uDC8F\uDC91\uDCAA\uDD7A\uDD95\uDD96\uDE4C\uDE4F\uDEC0\uDECC]|\uD83E[\uDD0C\uDD0F\uDD18-\uDD1C\uDD1E\uDD1F\uDD30-\uDD34\uDD36\uDD77\uDDB5\uDDB6\uDDBB\uDDD2\uDDD3\uDDD5])(?:\uD83C[\uDFFB-\uDFFF])|(?:[\u261D\u270C\u270D]|\uD83D[\uDD74\uDD90])(?:\uFE0F|\uD83C[\uDFFB-\uDFFF])|[\u270A\u270B]|\uD83C[\uDF85\uDFC2\uDFC7]|\uD83D[\uDC08\uDC15\uDC3B\uDC42\uDC43\uDC46-\uDC50\uDC66\uDC67\uDC6B-\uDC6D\uDC72\uDC74-\uDC76\uDC78\uDC7C\uDC83\uDC85\uDC8F\uDC91\uDCAA\uDD7A\uDD95\uDD96\uDE2E\uDE35\uDE36\uDE4C\uDE4F\uDEC0\uDECC]|\uD83E[\uDD0C\uDD0F\uDD18-\uDD1C\uDD1E\uDD1F\uDD30-\uDD34\uDD36\uDD77\uDDB5\uDDB6\uDDBB\uDDD2\uDDD3\uDDD5]|\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD]|\uD83D\uDC6F|\uD83E[\uDD3C\uDDDE\uDDDF]|[\u231A\u231B\u23E9-\u23EC\u23F0\u23F3\u25FD\u25FE\u2614\u2615\u2648-\u2653\u267F\u2693\u26A1\u26AA\u26AB\u26BD\u26BE\u26C4\u26C5\u26CE\u26D4\u26EA\u26F2\u26F3\u26F5\u26FA\u26FD\u2705\u2728\u274C\u274E\u2753-\u2755\u2757\u2795-\u2797\u27B0\u27BF\u2B1B\u2B1C\u2B50\u2B55]|\uD83C[\uDC04\uDCCF\uDD8E\uDD91-\uDD9A\uDE01\uDE1A\uDE2F\uDE32-\uDE36\uDE38-\uDE3A\uDE50\uDE51\uDF00-\uDF20\uDF2D-\uDF35\uDF37-\uDF7C\uDF7E-\uDF84\uDF86-\uDF93\uDFA0-\uDFC1\uDFC5\uDFC6\uDFC8\uDFC9\uDFCF-\uDFD3\uDFE0-\uDFF0\uDFF8-\uDFFF]|\uD83D[\uDC00-\uDC07\uDC09-\uDC14\uDC16-\uDC3A\uDC3C-\uDC3E\uDC40\uDC44\uDC45\uDC51-\uDC65\uDC6A\uDC79-\uDC7B\uDC7D-\uDC80\uDC84\uDC88-\uDC8E\uDC90\uDC92-\uDCA9\uDCAB-\uDCFC\uDCFF-\uDD3D\uDD4B-\uDD4E\uDD50-\uDD67\uDDA4\uDDFB-\uDE2D\uDE2F-\uDE34\uDE37-\uDE44\uDE48-\uDE4A\uDE80-\uDEA2\uDEA4-\uDEB3\uDEB7-\uDEBF\uDEC1-\uDEC5\uDED0-\uDED2\uDED5-\uDED7\uDEEB\uDEEC\uDEF4-\uDEFC\uDFE0-\uDFEB]|\uD83E[\uDD0D\uDD0E\uDD10-\uDD17\uDD1D\uDD20-\uDD25\uDD27-\uDD2F\uDD3A\uDD3F-\uDD45\uDD47-\uDD76\uDD78\uDD7A-\uDDB4\uDDB7\uDDBA\uDDBC-\uDDCB\uDDD0\uDDE0-\uDDFF\uDE70-\uDE74\uDE78-\uDE7A\uDE80-\uDE86\uDE90-\uDEA8\uDEB0-\uDEB6\uDEC0-\uDEC2\uDED0-\uDED6]|(?:[\u231A\u231B\u23E9-\u23EC\u23F0\u23F3\u25FD\u25FE\u2614\u2615\u2648-\u2653\u267F\u2693\u26A1\u26AA\u26AB\u26BD\u26BE\u26C4\u26C5\u26CE\u26D4\u26EA\u26F2\u26F3\u26F5\u26FA\u26FD\u2705\u270A\u270B\u2728\u274C\u274E\u2753-\u2755\u2757\u2795-\u2797\u27B0\u27BF\u2B1B\u2B1C\u2B50\u2B55]|\uD83C[\uDC04\uDCCF\uDD8E\uDD91-\uDD9A\uDDE6-\uDDFF\uDE01\uDE1A\uDE2F\uDE32-\uDE36\uDE38-\uDE3A\uDE50\uDE51\uDF00-\uDF20\uDF2D-\uDF35\uDF37-\uDF7C\uDF7E-\uDF93\uDFA0-\uDFCA\uDFCF-\uDFD3\uDFE0-\uDFF0\uDFF4\uDFF8-\uDFFF]|\uD83D[\uDC00-\uDC3E\uDC40\uDC42-\uDCFC\uDCFF-\uDD3D\uDD4B-\uDD4E\uDD50-\uDD67\uDD7A\uDD95\uDD96\uDDA4\uDDFB-\uDE4F\uDE80-\uDEC5\uDECC\uDED0-\uDED2\uDED5-\uDED7\uDEEB\uDEEC\uDEF4-\uDEFC\uDFE0-\uDFEB]|\uD83E[\uDD0C-\uDD3A\uDD3C-\uDD45\uDD47-\uDD78\uDD7A-\uDDCB\uDDCD-\uDDFF\uDE70-\uDE74\uDE78-\uDE7A\uDE80-\uDE86\uDE90-\uDEA8\uDEB0-\uDEB6\uDEC0-\uDEC2\uDED0-\uDED6])|(?:[#\*0-9\xA9\xAE\u203C\u2049\u2122\u2139\u2194-\u2199\u21A9\u21AA\u231A\u231B\u2328\u23CF\u23E9-\u23F3\u23F8-\u23FA\u24C2\u25AA\u25AB\u25B6\u25C0\u25FB-\u25FE\u2600-\u2604\u260E\u2611\u2614\u2615\u2618\u261D\u2620\u2622\u2623\u2626\u262A\u262E\u262F\u2638-\u263A\u2640\u2642\u2648-\u2653\u265F\u2660\u2663\u2665\u2666\u2668\u267B\u267E\u267F\u2692-\u2697\u2699\u269B\u269C\u26A0\u26A1\u26A7\u26AA\u26AB\u26B0\u26B1\u26BD\u26BE\u26C4\u26C5\u26C8\u26CE\u26CF\u26D1\u26D3\u26D4\u26E9\u26EA\u26F0-\u26F5\u26F7-\u26FA\u26FD\u2702\u2705\u2708-\u270D\u270F\u2712\u2714\u2716\u271D\u2721\u2728\u2733\u2734\u2744\u2747\u274C\u274E\u2753-\u2755\u2757\u2763\u2764\u2795-\u2797\u27A1\u27B0\u27BF\u2934\u2935\u2B05-\u2B07\u2B1B\u2B1C\u2B50\u2B55\u3030\u303D\u3297\u3299]|\uD83C[\uDC04\uDCCF\uDD70\uDD71\uDD7E\uDD7F\uDD8E\uDD91-\uDD9A\uDDE6-\uDDFF\uDE01\uDE02\uDE1A\uDE2F\uDE32-\uDE3A\uDE50\uDE51\uDF00-\uDF21\uDF24-\uDF93\uDF96\uDF97\uDF99-\uDF9B\uDF9E-\uDFF0\uDFF3-\uDFF5\uDFF7-\uDFFF]|\uD83D[\uDC00-\uDCFD\uDCFF-\uDD3D\uDD49-\uDD4E\uDD50-\uDD67\uDD6F\uDD70\uDD73-\uDD7A\uDD87\uDD8A-\uDD8D\uDD90\uDD95\uDD96\uDDA4\uDDA5\uDDA8\uDDB1\uDDB2\uDDBC\uDDC2-\uDDC4\uDDD1-\uDDD3\uDDDC-\uDDDE\uDDE1\uDDE3\uDDE8\uDDEF\uDDF3\uDDFA-\uDE4F\uDE80-\uDEC5\uDECB-\uDED2\uDED5-\uDED7\uDEE0-\uDEE5\uDEE9\uDEEB\uDEEC\uDEF0\uDEF3-\uDEFC\uDFE0-\uDFEB]|\uD83E[\uDD0C-\uDD3A\uDD3C-\uDD45\uDD47-\uDD78\uDD7A-\uDDCB\uDDCD-\uDDFF\uDE70-\uDE74\uDE78-\uDE7A\uDE80-\uDE86\uDE90-\uDEA8\uDEB0-\uDEB6\uDEC0-\uDEC2\uDED0-\uDED6])\uFE0F|(?:[\u261D\u26F9\u270A-\u270D]|\uD83C[\uDF85\uDFC2-\uDFC4\uDFC7\uDFCA-\uDFCC]|\uD83D[\uDC42\uDC43\uDC46-\uDC50\uDC66-\uDC78\uDC7C\uDC81-\uDC83\uDC85-\uDC87\uDC8F\uDC91\uDCAA\uDD74\uDD75\uDD7A\uDD90\uDD95\uDD96\uDE45-\uDE47\uDE4B-\uDE4F\uDEA3\uDEB4-\uDEB6\uDEC0\uDECC]|\uD83E[\uDD0C\uDD0F\uDD18-\uDD1F\uDD26\uDD30-\uDD39\uDD3C-\uDD3E\uDD77\uDDB5\uDDB6\uDDB8\uDDB9\uDDBB\uDDCD-\uDDCF\uDDD1-\uDDDD])/g;
};
const sD = L$1(FD);
function p(e, u = {}) {
	if (typeof e != "string" || e.length === 0 || (u = {
		ambiguousIsNarrow: !0,
		...u
	}, e = P$1(e), e.length === 0)) return 0;
	e = e.replace(sD(), "  ");
	const t = u.ambiguousIsNarrow ? 1 : 2;
	let F = 0;
	for (const s of e) {
		const i = s.codePointAt(0);
		if (i <= 31 || i >= 127 && i <= 159 || i >= 768 && i <= 879) continue;
		switch (eD.eastAsianWidth(s)) {
			case "F":
			case "W":
				F += 2;
				break;
			case "A":
				F += t;
				break;
			default: F += 1;
		}
	}
	return F;
}
const w = 10, N = (e = 0) => (u) => `\x1B[${u + e}m`, I = (e = 0) => (u) => `\x1B[${38 + e};5;${u}m`, R = (e = 0) => (u, t, F) => `\x1B[${38 + e};2;${u};${t};${F}m`, r = {
	modifier: {
		reset: [0, 0],
		bold: [1, 22],
		dim: [2, 22],
		italic: [3, 23],
		underline: [4, 24],
		overline: [53, 55],
		inverse: [7, 27],
		hidden: [8, 28],
		strikethrough: [9, 29]
	},
	color: {
		black: [30, 39],
		red: [31, 39],
		green: [32, 39],
		yellow: [33, 39],
		blue: [34, 39],
		magenta: [35, 39],
		cyan: [36, 39],
		white: [37, 39],
		blackBright: [90, 39],
		gray: [90, 39],
		grey: [90, 39],
		redBright: [91, 39],
		greenBright: [92, 39],
		yellowBright: [93, 39],
		blueBright: [94, 39],
		magentaBright: [95, 39],
		cyanBright: [96, 39],
		whiteBright: [97, 39]
	},
	bgColor: {
		bgBlack: [40, 49],
		bgRed: [41, 49],
		bgGreen: [42, 49],
		bgYellow: [43, 49],
		bgBlue: [44, 49],
		bgMagenta: [45, 49],
		bgCyan: [46, 49],
		bgWhite: [47, 49],
		bgBlackBright: [100, 49],
		bgGray: [100, 49],
		bgGrey: [100, 49],
		bgRedBright: [101, 49],
		bgGreenBright: [102, 49],
		bgYellowBright: [103, 49],
		bgBlueBright: [104, 49],
		bgMagentaBright: [105, 49],
		bgCyanBright: [106, 49],
		bgWhiteBright: [107, 49]
	}
};
Object.keys(r.modifier);
const iD = Object.keys(r.color), CD = Object.keys(r.bgColor);
[...iD, ...CD];
function rD() {
	const e = /* @__PURE__ */ new Map();
	for (const [u, t] of Object.entries(r)) {
		for (const [F, s] of Object.entries(t)) r[F] = {
			open: `\x1B[${s[0]}m`,
			close: `\x1B[${s[1]}m`
		}, t[F] = r[F], e.set(s[0], s[1]);
		Object.defineProperty(r, u, {
			value: t,
			enumerable: !1
		});
	}
	return Object.defineProperty(r, "codes", {
		value: e,
		enumerable: !1
	}), r.color.close = "\x1B[39m", r.bgColor.close = "\x1B[49m", r.color.ansi = N(), r.color.ansi256 = I(), r.color.ansi16m = R(), r.bgColor.ansi = N(w), r.bgColor.ansi256 = I(w), r.bgColor.ansi16m = R(w), Object.defineProperties(r, {
		rgbToAnsi256: {
			value: (u, t, F) => u === t && t === F ? u < 8 ? 16 : u > 248 ? 231 : Math.round((u - 8) / 247 * 24) + 232 : 16 + 36 * Math.round(u / 255 * 5) + 6 * Math.round(t / 255 * 5) + Math.round(F / 255 * 5),
			enumerable: !1
		},
		hexToRgb: {
			value: (u) => {
				const t = /[a-f\d]{6}|[a-f\d]{3}/i.exec(u.toString(16));
				if (!t) return [
					0,
					0,
					0
				];
				let [F] = t;
				F.length === 3 && (F = [...F].map((i) => i + i).join(""));
				const s = Number.parseInt(F, 16);
				return [
					s >> 16 & 255,
					s >> 8 & 255,
					s & 255
				];
			},
			enumerable: !1
		},
		hexToAnsi256: {
			value: (u) => r.rgbToAnsi256(...r.hexToRgb(u)),
			enumerable: !1
		},
		ansi256ToAnsi: {
			value: (u) => {
				if (u < 8) return 30 + u;
				if (u < 16) return 90 + (u - 8);
				let t, F, s;
				if (u >= 232) t = ((u - 232) * 10 + 8) / 255, F = t, s = t;
				else {
					u -= 16;
					const C = u % 36;
					t = Math.floor(u / 36) / 5, F = Math.floor(C / 6) / 5, s = C % 6 / 5;
				}
				const i = Math.max(t, F, s) * 2;
				if (i === 0) return 30;
				let D = 30 + (Math.round(s) << 2 | Math.round(F) << 1 | Math.round(t));
				return i === 2 && (D += 60), D;
			},
			enumerable: !1
		},
		rgbToAnsi: {
			value: (u, t, F) => r.ansi256ToAnsi(r.rgbToAnsi256(u, t, F)),
			enumerable: !1
		},
		hexToAnsi: {
			value: (u) => r.ansi256ToAnsi(r.hexToAnsi256(u)),
			enumerable: !1
		}
	}), r;
}
const ED = rD(), d$1 = new Set(["\x1B", ""]), oD = 39, y$1 = "\x07", V$1 = "[", nD = "]", G$1 = "m", _$1 = `${nD}8;;`, z = (e) => `${d$1.values().next().value}${V$1}${e}${G$1}`, K$1 = (e) => `${d$1.values().next().value}${_$1}${e}${y$1}`, aD = (e) => e.split(" ").map((u) => p(u)), k$1 = (e, u, t) => {
	const F = [...u];
	let s = !1, i = !1, D = p(P$1(e[e.length - 1]));
	for (const [C, n] of F.entries()) {
		const E = p(n);
		if (D + E <= t ? e[e.length - 1] += n : (e.push(n), D = 0), d$1.has(n) && (s = !0, i = F.slice(C + 1).join("").startsWith(_$1)), s) {
			i ? n === y$1 && (s = !1, i = !1) : n === G$1 && (s = !1);
			continue;
		}
		D += E, D === t && C < F.length - 1 && (e.push(""), D = 0);
	}
	!D && e[e.length - 1].length > 0 && e.length > 1 && (e[e.length - 2] += e.pop());
}, hD = (e) => {
	const u = e.split(" ");
	let t = u.length;
	for (; t > 0 && !(p(u[t - 1]) > 0);) t--;
	return t === u.length ? e : u.slice(0, t).join(" ") + u.slice(t).join("");
}, lD = (e, u, t = {}) => {
	if (t.trim !== !1 && e.trim() === "") return "";
	let F = "", s, i;
	const D = aD(e);
	let C = [""];
	for (const [E, a] of e.split(" ").entries()) {
		t.trim !== !1 && (C[C.length - 1] = C[C.length - 1].trimStart());
		let o = p(C[C.length - 1]);
		if (E !== 0 && (o >= u && (t.wordWrap === !1 || t.trim === !1) && (C.push(""), o = 0), (o > 0 || t.trim === !1) && (C[C.length - 1] += " ", o++)), t.hard && D[E] > u) {
			const c = u - o, f = 1 + Math.floor((D[E] - c - 1) / u);
			Math.floor((D[E] - 1) / u) < f && C.push(""), k$1(C, a, u);
			continue;
		}
		if (o + D[E] > u && o > 0 && D[E] > 0) {
			if (t.wordWrap === !1 && o < u) {
				k$1(C, a, u);
				continue;
			}
			C.push("");
		}
		if (o + D[E] > u && t.wordWrap === !1) {
			k$1(C, a, u);
			continue;
		}
		C[C.length - 1] += a;
	}
	t.trim !== !1 && (C = C.map((E) => hD(E)));
	const n = [...C.join(`
`)];
	for (const [E, a] of n.entries()) {
		if (F += a, d$1.has(a)) {
			const { groups: c } = new RegExp(`(?:\\${V$1}(?<code>\\d+)m|\\${_$1}(?<uri>.*)${y$1})`).exec(n.slice(E).join("")) || { groups: {} };
			if (c.code !== void 0) {
				const f = Number.parseFloat(c.code);
				s = f === oD ? void 0 : f;
			} else c.uri !== void 0 && (i = c.uri.length === 0 ? void 0 : c.uri);
		}
		const o = ED.codes.get(Number(s));
		n[E + 1] === `
` ? (i && (F += K$1("")), s && o && (F += z(o))) : a === `
` && (s && o && (F += z(s)), i && (F += K$1(i)));
	}
	return F;
};
function Y$1(e, u, t) {
	return String(e).normalize().replace(/\r\n/g, `
`).split(`
`).map((F) => lD(F, u, t)).join(`
`);
}
const B = {
	actions: new Set([
		"up",
		"down",
		"left",
		"right",
		"space",
		"enter",
		"cancel"
	]),
	aliases: new Map([
		["k", "up"],
		["j", "down"],
		["h", "left"],
		["l", "right"],
		["", "cancel"],
		["escape", "cancel"]
	])
};
function $(e, u) {
	if (typeof e == "string") return B.aliases.get(e) === u;
	for (const t of e) if (t !== void 0 && $(t, u)) return !0;
	return !1;
}
function BD(e, u) {
	if (e === u) return;
	const t = e.split(`
`), F = u.split(`
`), s = [];
	for (let i = 0; i < Math.max(t.length, F.length); i++) t[i] !== F[i] && s.push(i);
	return s;
}
const AD = globalThis.process.platform.startsWith("win"), S = Symbol("clack:cancel");
function pD(e) {
	return e === S;
}
function m(e, u) {
	const t = e;
	t.isTTY && t.setRawMode(u);
}
function fD({ input: e = stdin, output: u = stdout, overwrite: t = !0, hideCursor: F = !0 } = {}) {
	const s = g.createInterface({
		input: e,
		output: u,
		prompt: "",
		tabSize: 1
	});
	g.emitKeypressEvents(e, s), e.isTTY && e.setRawMode(!0);
	const i = (D, { name: C, sequence: n }) => {
		if ($([
			String(D),
			C,
			n
		], "cancel")) {
			F && u.write(import_src.cursor.show), process.exit(0);
			return;
		}
		if (!t) return;
		const a = C === "return" ? 0 : -1, o = C === "return" ? -1 : 0;
		g.moveCursor(u, a, o, () => {
			g.clearLine(u, 1, () => {
				e.once("keypress", i);
			});
		});
	};
	return F && u.write(import_src.cursor.hide), e.once("keypress", i), () => {
		e.off("keypress", i), F && u.write(import_src.cursor.show), e.isTTY && !AD && e.setRawMode(!1), s.terminal = !1, s.close();
	};
}
var gD = Object.defineProperty, vD = (e, u, t) => u in e ? gD(e, u, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: t
}) : e[u] = t, h = (e, u, t) => (vD(e, typeof u != "symbol" ? u + "" : u, t), t);
var x$1 = class {
	constructor(u, t = !0) {
		h(this, "input"), h(this, "output"), h(this, "_abortSignal"), h(this, "rl"), h(this, "opts"), h(this, "_render"), h(this, "_track", !1), h(this, "_prevFrame", ""), h(this, "_subscribers", /* @__PURE__ */ new Map()), h(this, "_cursor", 0), h(this, "state", "initial"), h(this, "error", ""), h(this, "value");
		const { input: F = stdin, output: s = stdout, render: i, signal: D, ...C } = u;
		this.opts = C, this.onKeypress = this.onKeypress.bind(this), this.close = this.close.bind(this), this.render = this.render.bind(this), this._render = i.bind(this), this._track = t, this._abortSignal = D, this.input = F, this.output = s;
	}
	unsubscribe() {
		this._subscribers.clear();
	}
	setSubscriber(u, t) {
		const F = this._subscribers.get(u) ?? [];
		F.push(t), this._subscribers.set(u, F);
	}
	on(u, t) {
		this.setSubscriber(u, { cb: t });
	}
	once(u, t) {
		this.setSubscriber(u, {
			cb: t,
			once: !0
		});
	}
	emit(u, ...t) {
		const F = this._subscribers.get(u) ?? [], s = [];
		for (const i of F) i.cb(...t), i.once && s.push(() => F.splice(F.indexOf(i), 1));
		for (const i of s) i();
	}
	prompt() {
		return new Promise((u, t) => {
			if (this._abortSignal) {
				if (this._abortSignal.aborted) return this.state = "cancel", this.close(), u(S);
				this._abortSignal.addEventListener("abort", () => {
					this.state = "cancel", this.close();
				}, { once: !0 });
			}
			const F = new Writable();
			F._write = (s, i, D) => {
				this._track && (this.value = this.rl?.line.replace(/\t/g, ""), this._cursor = this.rl?.cursor ?? 0, this.emit("value", this.value)), D();
			}, this.input.pipe(F), this.rl = O.createInterface({
				input: this.input,
				output: F,
				tabSize: 2,
				prompt: "",
				escapeCodeTimeout: 50,
				terminal: !0
			}), O.emitKeypressEvents(this.input, this.rl), this.rl.prompt(), this.opts.initialValue !== void 0 && this._track && this.rl.write(this.opts.initialValue), this.input.on("keypress", this.onKeypress), m(this.input, !0), this.output.on("resize", this.render), this.render(), this.once("submit", () => {
				this.output.write(import_src.cursor.show), this.output.off("resize", this.render), m(this.input, !1), u(this.value);
			}), this.once("cancel", () => {
				this.output.write(import_src.cursor.show), this.output.off("resize", this.render), m(this.input, !1), u(S);
			});
		});
	}
	onKeypress(u, t) {
		if (this.state === "error" && (this.state = "active"), t?.name && (!this._track && B.aliases.has(t.name) && this.emit("cursor", B.aliases.get(t.name)), B.actions.has(t.name) && this.emit("cursor", t.name)), u && (u.toLowerCase() === "y" || u.toLowerCase() === "n") && this.emit("confirm", u.toLowerCase() === "y"), u === "	" && this.opts.placeholder && (this.value || (this.rl?.write(this.opts.placeholder), this.emit("value", this.opts.placeholder))), u && this.emit("key", u.toLowerCase()), t?.name === "return") {
			if (this.opts.validate) {
				const F = this.opts.validate(this.value);
				F && (this.error = F instanceof Error ? F.message : F, this.state = "error", this.rl?.write(this.value));
			}
			this.state !== "error" && (this.state = "submit");
		}
		$([
			u,
			t?.name,
			t?.sequence
		], "cancel") && (this.state = "cancel"), (this.state === "submit" || this.state === "cancel") && this.emit("finalize"), this.render(), (this.state === "submit" || this.state === "cancel") && this.close();
	}
	close() {
		this.input.unpipe(), this.input.removeListener("keypress", this.onKeypress), this.output.write(`
`), m(this.input, !1), this.rl?.close(), this.rl = void 0, this.emit(`${this.state}`, this.value), this.unsubscribe();
	}
	restoreCursor() {
		const u = Y$1(this._prevFrame, process.stdout.columns, { hard: !0 }).split(`
`).length - 1;
		this.output.write(import_src.cursor.move(-999, u * -1));
	}
	render() {
		const u = Y$1(this._render(this) ?? "", process.stdout.columns, { hard: !0 });
		if (u !== this._prevFrame) {
			if (this.state === "initial") this.output.write(import_src.cursor.hide);
			else {
				const t = BD(this._prevFrame, u);
				if (this.restoreCursor(), t && t?.length === 1) {
					const F = t[0];
					this.output.write(import_src.cursor.move(0, F)), this.output.write(import_src.erase.lines(1));
					const s = u.split(`
`);
					this.output.write(s[F]), this._prevFrame = u, this.output.write(import_src.cursor.move(0, s.length - F - 1));
					return;
				}
				if (t && t?.length > 1) {
					const F = t[0];
					this.output.write(import_src.cursor.move(0, F)), this.output.write(import_src.erase.down());
					const s = u.split(`
`).slice(F);
					this.output.write(s.join(`
`)), this._prevFrame = u;
					return;
				}
				this.output.write(import_src.erase.down());
			}
			this.output.write(u), this.state === "initial" && (this.state = "active"), this._prevFrame = u;
		}
	}
};
var dD = class extends x$1 {
	get cursor() {
		return this.value ? 0 : 1;
	}
	get _value() {
		return this.cursor === 0;
	}
	constructor(u) {
		super(u, !1), this.value = !!u.initialValue, this.on("value", () => {
			this.value = this._value;
		}), this.on("confirm", (t) => {
			this.output.write(import_src.cursor.move(0, -1)), this.value = t, this.state = "submit", this.close();
		}), this.on("cursor", () => {
			this.value = !this.value;
		});
	}
};
var mD = Object.defineProperty, bD = (e, u, t) => u in e ? mD(e, u, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: t
}) : e[u] = t, Z = (e, u, t) => (bD(e, typeof u != "symbol" ? u + "" : u, t), t), q$1 = (e, u, t) => {
	if (!u.has(e)) throw TypeError("Cannot " + t);
}, T$1 = (e, u, t) => (q$1(e, u, "read from private field"), t ? t.call(e) : u.get(e)), wD = (e, u, t) => {
	if (u.has(e)) throw TypeError("Cannot add the same private member more than once");
	u instanceof WeakSet ? u.add(e) : u.set(e, t);
}, yD = (e, u, t, F) => (q$1(e, u, "write to private field"), F ? F.call(e, t) : u.set(e, t), t), A$1;
let _D = class extends x$1 {
	constructor(u) {
		super(u, !1), Z(this, "options"), Z(this, "cursor", 0), wD(this, A$1, void 0);
		const { options: t } = u;
		yD(this, A$1, u.selectableGroups !== !1), this.options = Object.entries(t).flatMap(([F, s]) => [{
			value: F,
			group: !0,
			label: F
		}, ...s.map((i) => ({
			...i,
			group: F
		}))]), this.value = [...u.initialValues ?? []], this.cursor = Math.max(this.options.findIndex(({ value: F }) => F === u.cursorAt), T$1(this, A$1) ? 0 : 1), this.on("cursor", (F) => {
			switch (F) {
				case "left":
				case "up": {
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					const s = this.options[this.cursor]?.group === !0;
					!T$1(this, A$1) && s && (this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1);
					break;
				}
				case "down":
				case "right": {
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					const s = this.options[this.cursor]?.group === !0;
					!T$1(this, A$1) && s && (this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1);
					break;
				}
				case "space":
					this.toggleValue();
					break;
			}
		});
	}
	getGroupItems(u) {
		return this.options.filter((t) => t.group === u);
	}
	isGroupSelected(u) {
		return this.getGroupItems(u).every((t) => this.value.includes(t.value));
	}
	toggleValue() {
		const u = this.options[this.cursor];
		if (u.group === !0) {
			const t = u.value, F = this.getGroupItems(t);
			this.isGroupSelected(t) ? this.value = this.value.filter((s) => F.findIndex((i) => i.value === s) === -1) : this.value = [...this.value, ...F.map((s) => s.value)], this.value = Array.from(new Set(this.value));
		} else this.value = this.value.includes(u.value) ? this.value.filter((F) => F !== u.value) : [...this.value, u.value];
	}
};
A$1 = /* @__PURE__ */ new WeakMap();
var kD = Object.defineProperty, $D = (e, u, t) => u in e ? kD(e, u, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: t
}) : e[u] = t, H = (e, u, t) => ($D(e, typeof u != "symbol" ? u + "" : u, t), t);
let SD = class extends x$1 {
	constructor(u) {
		super(u, !1), H(this, "options"), H(this, "cursor", 0), this.options = u.options, this.value = [...u.initialValues ?? []], this.cursor = Math.max(this.options.findIndex(({ value: t }) => t === u.cursorAt), 0), this.on("key", (t) => {
			t === "a" && this.toggleAll();
		}), this.on("cursor", (t) => {
			switch (t) {
				case "left":
				case "up":
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					break;
				case "down":
				case "right":
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					break;
				case "space":
					this.toggleValue();
					break;
			}
		});
	}
	get _value() {
		return this.options[this.cursor].value;
	}
	toggleAll() {
		this.value = this.value.length === this.options.length ? [] : this.options.map((t) => t.value);
	}
	toggleValue() {
		this.value = this.value.includes(this._value) ? this.value.filter((t) => t !== this._value) : [...this.value, this._value];
	}
};
var TD = Object.defineProperty, jD = (e, u, t) => u in e ? TD(e, u, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: t
}) : e[u] = t, U$1 = (e, u, t) => (jD(e, typeof u != "symbol" ? u + "" : u, t), t);
var MD = class extends x$1 {
	constructor({ mask: u, ...t }) {
		super(t), U$1(this, "valueWithCursor", ""), U$1(this, "_mask", "•"), this._mask = u ?? "•", this.on("finalize", () => {
			this.valueWithCursor = this.masked;
		}), this.on("value", () => {
			if (this.cursor >= this.value.length) this.valueWithCursor = `${this.masked}${import_picocolors.default.inverse(import_picocolors.default.hidden("_"))}`;
			else {
				const F = this.masked.slice(0, this.cursor), s = this.masked.slice(this.cursor);
				this.valueWithCursor = `${F}${import_picocolors.default.inverse(s[0])}${s.slice(1)}`;
			}
		});
	}
	get cursor() {
		return this._cursor;
	}
	get masked() {
		return this.value.replaceAll(/./g, this._mask);
	}
};
var OD = Object.defineProperty, PD = (e, u, t) => u in e ? OD(e, u, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: t
}) : e[u] = t, J$1 = (e, u, t) => (PD(e, typeof u != "symbol" ? u + "" : u, t), t);
var LD = class extends x$1 {
	constructor(u) {
		super(u, !1), J$1(this, "options"), J$1(this, "cursor", 0), this.options = u.options, this.cursor = this.options.findIndex(({ value: t }) => t === u.initialValue), this.cursor === -1 && (this.cursor = 0), this.changeValue(), this.on("cursor", (t) => {
			switch (t) {
				case "left":
				case "up":
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					break;
				case "down":
				case "right":
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					break;
			}
			this.changeValue();
		});
	}
	get _value() {
		return this.options[this.cursor];
	}
	changeValue() {
		this.value = this._value.value;
	}
};
var WD = Object.defineProperty, ND = (e, u, t) => u in e ? WD(e, u, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: t
}) : e[u] = t, Q = (e, u, t) => (ND(e, typeof u != "symbol" ? u + "" : u, t), t);
var ID = class extends x$1 {
	constructor(u) {
		super(u, !1), Q(this, "options"), Q(this, "cursor", 0), this.options = u.options;
		const t = this.options.map(({ value: [F] }) => F?.toLowerCase());
		this.cursor = Math.max(t.indexOf(u.initialValue), 0), this.on("key", (F) => {
			if (!t.includes(F)) return;
			const s = this.options.find(({ value: [i] }) => i?.toLowerCase() === F);
			s && (this.value = s.value, this.state = "submit", this.emit("submit"));
		});
	}
};
var RD = class extends x$1 {
	get valueWithCursor() {
		if (this.state === "submit") return this.value;
		if (this.cursor >= this.value.length) return `${this.value}\u2588`;
		const u = this.value.slice(0, this.cursor), [t, ...F] = this.value.slice(this.cursor);
		return `${u}${import_picocolors.default.inverse(t)}${F.join("")}`;
	}
	get cursor() {
		return this._cursor;
	}
	constructor(u) {
		super(u), this.on("finalize", () => {
			this.value || (this.value = u.defaultValue);
		});
	}
};

//#endregion
//#region node_modules/.pnpm/@clack+prompts@0.11.0/node_modules/@clack/prompts/dist/index.mjs
function ce() {
	return y.platform !== "win32" ? y.env.TERM !== "linux" : !!y.env.CI || !!y.env.WT_SESSION || !!y.env.TERMINUS_SUBLIME || y.env.ConEmuTask === "{cmd::Cmder}" || y.env.TERM_PROGRAM === "Terminus-Sublime" || y.env.TERM_PROGRAM === "vscode" || y.env.TERM === "xterm-256color" || y.env.TERM === "alacritty" || y.env.TERMINAL_EMULATOR === "JetBrains-JediTerm";
}
const V = ce(), u = (t, n) => V ? t : n, le = u("◆", "*"), L = u("■", "x"), W = u("▲", "x"), C = u("◇", "o"), ue = u("┌", "T"), o = u("│", "|"), d = u("└", "—"), k = u("●", ">"), P = u("○", " "), A = u("◻", "[•]"), T = u("◼", "[+]"), F = u("◻", "[ ]"), $e = u("▪", "•"), _ = u("─", "-"), me = u("╮", "+"), de = u("├", "+"), pe = u("╯", "+"), q = u("●", "•"), D = u("◆", "*"), U = u("▲", "!"), K = u("■", "x"), b = (t) => {
	switch (t) {
		case "initial":
		case "active": return import_picocolors.default.cyan(le);
		case "cancel": return import_picocolors.default.red(L);
		case "error": return import_picocolors.default.yellow(W);
		case "submit": return import_picocolors.default.green(C);
	}
}, G = (t) => {
	const { cursor: n, options: r, style: i } = t, s = t.maxItems ?? Number.POSITIVE_INFINITY, c = Math.max(process.stdout.rows - 4, 0), a = Math.min(c, Math.max(s, 5));
	let l = 0;
	n >= l + a - 3 ? l = Math.max(Math.min(n - a + 3, r.length - a), 0) : n < l + 2 && (l = Math.max(n - 2, 0));
	const $ = a < r.length && l > 0, g = a < r.length && l + a < r.length;
	return r.slice(l, l + a).map((p, v, f) => {
		const j = v === 0 && $, E = v === f.length - 1 && g;
		return j || E ? import_picocolors.default.dim("...") : i(p, v + l === n);
	});
}, he = (t) => new RD({
	validate: t.validate,
	placeholder: t.placeholder,
	defaultValue: t.defaultValue,
	initialValue: t.initialValue,
	render() {
		const n = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`, r = t.placeholder ? import_picocolors.default.inverse(t.placeholder[0]) + import_picocolors.default.dim(t.placeholder.slice(1)) : import_picocolors.default.inverse(import_picocolors.default.hidden("_")), i = this.value ? this.valueWithCursor : r;
		switch (this.state) {
			case "error": return `${n.trim()}
${import_picocolors.default.yellow(o)}  ${i}
${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(this.error)}
`;
			case "submit": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(this.value || t.placeholder)}`;
			case "cancel": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(this.value ?? ""))}${this.value?.trim() ? `
${import_picocolors.default.gray(o)}` : ""}`;
			default: return `${n}${import_picocolors.default.cyan(o)}  ${i}
${import_picocolors.default.cyan(d)}
`;
		}
	}
}).prompt(), ge = (t) => new MD({
	validate: t.validate,
	mask: t.mask ?? $e,
	render() {
		const n = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`, r = this.valueWithCursor, i = this.masked;
		switch (this.state) {
			case "error": return `${n.trim()}
${import_picocolors.default.yellow(o)}  ${i}
${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(this.error)}
`;
			case "submit": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(i)}`;
			case "cancel": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(i ?? ""))}${i ? `
${import_picocolors.default.gray(o)}` : ""}`;
			default: return `${n}${import_picocolors.default.cyan(o)}  ${r}
${import_picocolors.default.cyan(d)}
`;
		}
	}
}).prompt(), ye = (t) => {
	const n = t.active ?? "Yes", r = t.inactive ?? "No";
	return new dD({
		active: n,
		inactive: r,
		initialValue: t.initialValue ?? !0,
		render() {
			const i = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`, s = this.value ? n : r;
			switch (this.state) {
				case "submit": return `${i}${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(s)}`;
				case "cancel": return `${i}${import_picocolors.default.gray(o)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(s))}
${import_picocolors.default.gray(o)}`;
				default: return `${i}${import_picocolors.default.cyan(o)}  ${this.value ? `${import_picocolors.default.green(k)} ${n}` : `${import_picocolors.default.dim(P)} ${import_picocolors.default.dim(n)}`} ${import_picocolors.default.dim("/")} ${this.value ? `${import_picocolors.default.dim(P)} ${import_picocolors.default.dim(r)}` : `${import_picocolors.default.green(k)} ${r}`}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, ve = (t) => {
	const n = (r, i) => {
		const s = r.label ?? String(r.value);
		switch (i) {
			case "selected": return `${import_picocolors.default.dim(s)}`;
			case "active": return `${import_picocolors.default.green(k)} ${s} ${r.hint ? import_picocolors.default.dim(`(${r.hint})`) : ""}`;
			case "cancelled": return `${import_picocolors.default.strikethrough(import_picocolors.default.dim(s))}`;
			default: return `${import_picocolors.default.dim(P)} ${import_picocolors.default.dim(s)}`;
		}
	};
	return new LD({
		options: t.options,
		initialValue: t.initialValue,
		render() {
			const r = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`;
			switch (this.state) {
				case "submit": return `${r}${import_picocolors.default.gray(o)}  ${n(this.options[this.cursor], "selected")}`;
				case "cancel": return `${r}${import_picocolors.default.gray(o)}  ${n(this.options[this.cursor], "cancelled")}
${import_picocolors.default.gray(o)}`;
				default: return `${r}${import_picocolors.default.cyan(o)}  ${G({
					cursor: this.cursor,
					options: this.options,
					maxItems: t.maxItems,
					style: (i, s) => n(i, s ? "active" : "inactive")
				}).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, we = (t) => {
	const n = (r, i = "inactive") => {
		const s = r.label ?? String(r.value);
		return i === "selected" ? `${import_picocolors.default.dim(s)}` : i === "cancelled" ? `${import_picocolors.default.strikethrough(import_picocolors.default.dim(s))}` : i === "active" ? `${import_picocolors.default.bgCyan(import_picocolors.default.gray(` ${r.value} `))} ${s} ${r.hint ? import_picocolors.default.dim(`(${r.hint})`) : ""}` : `${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(` ${r.value} `)))} ${s} ${r.hint ? import_picocolors.default.dim(`(${r.hint})`) : ""}`;
	};
	return new ID({
		options: t.options,
		initialValue: t.initialValue,
		render() {
			const r = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`;
			switch (this.state) {
				case "submit": return `${r}${import_picocolors.default.gray(o)}  ${n(this.options.find((i) => i.value === this.value) ?? t.options[0], "selected")}`;
				case "cancel": return `${r}${import_picocolors.default.gray(o)}  ${n(this.options[0], "cancelled")}
${import_picocolors.default.gray(o)}`;
				default: return `${r}${import_picocolors.default.cyan(o)}  ${this.options.map((i, s) => n(i, s === this.cursor ? "active" : "inactive")).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, fe = (t) => {
	const n = (r, i) => {
		const s = r.label ?? String(r.value);
		return i === "active" ? `${import_picocolors.default.cyan(A)} ${s} ${r.hint ? import_picocolors.default.dim(`(${r.hint})`) : ""}` : i === "selected" ? `${import_picocolors.default.green(T)} ${import_picocolors.default.dim(s)} ${r.hint ? import_picocolors.default.dim(`(${r.hint})`) : ""}` : i === "cancelled" ? `${import_picocolors.default.strikethrough(import_picocolors.default.dim(s))}` : i === "active-selected" ? `${import_picocolors.default.green(T)} ${s} ${r.hint ? import_picocolors.default.dim(`(${r.hint})`) : ""}` : i === "submitted" ? `${import_picocolors.default.dim(s)}` : `${import_picocolors.default.dim(F)} ${import_picocolors.default.dim(s)}`;
	};
	return new SD({
		options: t.options,
		initialValues: t.initialValues,
		required: t.required ?? !0,
		cursorAt: t.cursorAt,
		validate(r) {
			if (this.required && r.length === 0) return `Please select at least one option.
${import_picocolors.default.reset(import_picocolors.default.dim(`Press ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" space ")))} to select, ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" enter ")))} to submit`))}`;
		},
		render() {
			const r = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`, i = (s, c) => {
				const a = this.value.includes(s.value);
				return c && a ? n(s, "active-selected") : a ? n(s, "selected") : n(s, c ? "active" : "inactive");
			};
			switch (this.state) {
				case "submit": return `${r}${import_picocolors.default.gray(o)}  ${this.options.filter(({ value: s }) => this.value.includes(s)).map((s) => n(s, "submitted")).join(import_picocolors.default.dim(", ")) || import_picocolors.default.dim("none")}`;
				case "cancel": {
					const s = this.options.filter(({ value: c }) => this.value.includes(c)).map((c) => n(c, "cancelled")).join(import_picocolors.default.dim(", "));
					return `${r}${import_picocolors.default.gray(o)}  ${s.trim() ? `${s}
${import_picocolors.default.gray(o)}` : ""}`;
				}
				case "error": {
					const s = this.error.split(`
`).map((c, a) => a === 0 ? `${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(c)}` : `   ${c}`).join(`
`);
					return `${r + import_picocolors.default.yellow(o)}  ${G({
						options: this.options,
						cursor: this.cursor,
						maxItems: t.maxItems,
						style: i
					}).join(`
${import_picocolors.default.yellow(o)}  `)}
${s}
`;
				}
				default: return `${r}${import_picocolors.default.cyan(o)}  ${G({
					options: this.options,
					cursor: this.cursor,
					maxItems: t.maxItems,
					style: i
				}).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, be = (t) => {
	const { selectableGroups: n = !0 } = t, r = (i, s, c = []) => {
		const a = i.label ?? String(i.value), l = typeof i.group == "string", $ = l && (c[c.indexOf(i) + 1] ?? { group: !0 }), g = l && $.group === !0, p = l ? n ? `${g ? d : o} ` : "  " : "";
		if (s === "active") return `${import_picocolors.default.dim(p)}${import_picocolors.default.cyan(A)} ${a} ${i.hint ? import_picocolors.default.dim(`(${i.hint})`) : ""}`;
		if (s === "group-active") return `${p}${import_picocolors.default.cyan(A)} ${import_picocolors.default.dim(a)}`;
		if (s === "group-active-selected") return `${p}${import_picocolors.default.green(T)} ${import_picocolors.default.dim(a)}`;
		if (s === "selected") {
			const f = l || n ? import_picocolors.default.green(T) : "";
			return `${import_picocolors.default.dim(p)}${f} ${import_picocolors.default.dim(a)} ${i.hint ? import_picocolors.default.dim(`(${i.hint})`) : ""}`;
		}
		if (s === "cancelled") return `${import_picocolors.default.strikethrough(import_picocolors.default.dim(a))}`;
		if (s === "active-selected") return `${import_picocolors.default.dim(p)}${import_picocolors.default.green(T)} ${a} ${i.hint ? import_picocolors.default.dim(`(${i.hint})`) : ""}`;
		if (s === "submitted") return `${import_picocolors.default.dim(a)}`;
		const v = l || n ? import_picocolors.default.dim(F) : "";
		return `${import_picocolors.default.dim(p)}${v} ${import_picocolors.default.dim(a)}`;
	};
	return new _D({
		options: t.options,
		initialValues: t.initialValues,
		required: t.required ?? !0,
		cursorAt: t.cursorAt,
		selectableGroups: n,
		validate(i) {
			if (this.required && i.length === 0) return `Please select at least one option.
${import_picocolors.default.reset(import_picocolors.default.dim(`Press ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" space ")))} to select, ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" enter ")))} to submit`))}`;
		},
		render() {
			const i = `${import_picocolors.default.gray(o)}
${b(this.state)}  ${t.message}
`;
			switch (this.state) {
				case "submit": return `${i}${import_picocolors.default.gray(o)}  ${this.options.filter(({ value: s }) => this.value.includes(s)).map((s) => r(s, "submitted")).join(import_picocolors.default.dim(", "))}`;
				case "cancel": {
					const s = this.options.filter(({ value: c }) => this.value.includes(c)).map((c) => r(c, "cancelled")).join(import_picocolors.default.dim(", "));
					return `${i}${import_picocolors.default.gray(o)}  ${s.trim() ? `${s}
${import_picocolors.default.gray(o)}` : ""}`;
				}
				case "error": {
					const s = this.error.split(`
`).map((c, a) => a === 0 ? `${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(c)}` : `   ${c}`).join(`
`);
					return `${i}${import_picocolors.default.yellow(o)}  ${this.options.map((c, a, l) => {
						const $ = this.value.includes(c.value) || c.group === !0 && this.isGroupSelected(`${c.value}`), g = a === this.cursor;
						return !g && typeof c.group == "string" && this.options[this.cursor].value === c.group ? r(c, $ ? "group-active-selected" : "group-active", l) : g && $ ? r(c, "active-selected", l) : $ ? r(c, "selected", l) : r(c, g ? "active" : "inactive", l);
					}).join(`
${import_picocolors.default.yellow(o)}  `)}
${s}
`;
				}
				default: return `${i}${import_picocolors.default.cyan(o)}  ${this.options.map((s, c, a) => {
					const l = this.value.includes(s.value) || s.group === !0 && this.isGroupSelected(`${s.value}`), $ = c === this.cursor;
					return !$ && typeof s.group == "string" && this.options[this.cursor].value === s.group ? r(s, l ? "group-active-selected" : "group-active", a) : $ && l ? r(s, "active-selected", a) : l ? r(s, "selected", a) : r(s, $ ? "active" : "inactive", a);
				}).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, Me = (t = "", n = "") => {
	const r = `
${t}
`.split(`
`), i = stripVTControlCharacters(n).length, s = Math.max(r.reduce((a, l) => {
		const $ = stripVTControlCharacters(l);
		return $.length > a ? $.length : a;
	}, 0), i) + 2, c = r.map((a) => `${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(a)}${" ".repeat(s - stripVTControlCharacters(a).length)}${import_picocolors.default.gray(o)}`).join(`
`);
	process.stdout.write(`${import_picocolors.default.gray(o)}
${import_picocolors.default.green(C)}  ${import_picocolors.default.reset(n)} ${import_picocolors.default.gray(_.repeat(Math.max(s - i - 1, 1)) + me)}
${c}
${import_picocolors.default.gray(de + _.repeat(s + 2) + pe)}
`);
}, xe = (t = "") => {
	process.stdout.write(`${import_picocolors.default.gray(d)}  ${import_picocolors.default.red(t)}

`);
}, Ie = (t = "") => {
	process.stdout.write(`${import_picocolors.default.gray(ue)}  ${t}
`);
}, Se = (t = "") => {
	process.stdout.write(`${import_picocolors.default.gray(o)}
${import_picocolors.default.gray(d)}  ${t}

`);
}, M = {
	message: (t = "", { symbol: n = import_picocolors.default.gray(o) } = {}) => {
		const r = [`${import_picocolors.default.gray(o)}`];
		if (t) {
			const [i, ...s] = t.split(`
`);
			r.push(`${n}  ${i}`, ...s.map((c) => `${import_picocolors.default.gray(o)}  ${c}`));
		}
		process.stdout.write(`${r.join(`
`)}
`);
	},
	info: (t) => {
		M.message(t, { symbol: import_picocolors.default.blue(q) });
	},
	success: (t) => {
		M.message(t, { symbol: import_picocolors.default.green(D) });
	},
	step: (t) => {
		M.message(t, { symbol: import_picocolors.default.green(C) });
	},
	warn: (t) => {
		M.message(t, { symbol: import_picocolors.default.yellow(U) });
	},
	warning: (t) => {
		M.warn(t);
	},
	error: (t) => {
		M.message(t, { symbol: import_picocolors.default.red(K) });
	}
}, J = `${import_picocolors.default.gray(o)}  `, x = {
	message: async (t, { symbol: n = import_picocolors.default.gray(o) } = {}) => {
		process.stdout.write(`${import_picocolors.default.gray(o)}
${n}  `);
		let r = 3;
		for await (let i of t) {
			i = i.replace(/\n/g, `
${J}`), i.includes(`
`) && (r = 3 + stripVTControlCharacters(i.slice(i.lastIndexOf(`
`))).length);
			const s = stripVTControlCharacters(i).length;
			r + s < process.stdout.columns ? (r += s, process.stdout.write(i)) : (process.stdout.write(`
${J}${i.trimStart()}`), r = 3 + stripVTControlCharacters(i.trimStart()).length);
		}
		process.stdout.write(`
`);
	},
	info: (t) => x.message(t, { symbol: import_picocolors.default.blue(q) }),
	success: (t) => x.message(t, { symbol: import_picocolors.default.green(D) }),
	step: (t) => x.message(t, { symbol: import_picocolors.default.green(C) }),
	warn: (t) => x.message(t, { symbol: import_picocolors.default.yellow(U) }),
	warning: (t) => x.warn(t),
	error: (t) => x.message(t, { symbol: import_picocolors.default.red(K) })
}, Y = ({ indicator: t = "dots" } = {}) => {
	const n = V ? [
		"◒",
		"◐",
		"◓",
		"◑"
	] : [
		"•",
		"o",
		"O",
		"0"
	], r = V ? 80 : 120, i = process.env.CI === "true";
	let s, c, a = !1, l = "", $, g = performance.now();
	const p = (m) => {
		a && N(m > 1 ? "Something went wrong" : "Canceled", m);
	}, v = () => p(2), f = () => p(1), j = () => {
		process.on("uncaughtExceptionMonitor", v), process.on("unhandledRejection", v), process.on("SIGINT", f), process.on("SIGTERM", f), process.on("exit", p);
	}, E = () => {
		process.removeListener("uncaughtExceptionMonitor", v), process.removeListener("unhandledRejection", v), process.removeListener("SIGINT", f), process.removeListener("SIGTERM", f), process.removeListener("exit", p);
	}, B = () => {
		if ($ === void 0) return;
		i && process.stdout.write(`
`);
		const m = $.split(`
`);
		process.stdout.write(import_src.cursor.move(-999, m.length - 1)), process.stdout.write(import_src.erase.down(m.length));
	}, R = (m) => m.replace(/\.+$/, ""), O = (m) => {
		const h = (performance.now() - m) / 1e3, w = Math.floor(h / 60), I = Math.floor(h % 60);
		return w > 0 ? `[${w}m ${I}s]` : `[${I}s]`;
	}, H = (m = "") => {
		a = !0, s = fD(), l = R(m), g = performance.now(), process.stdout.write(`${import_picocolors.default.gray(o)}
`);
		let h = 0, w = 0;
		j(), c = setInterval(() => {
			if (i && l === $) return;
			B(), $ = l;
			const I = import_picocolors.default.magenta(n[h]);
			if (i) process.stdout.write(`${I}  ${l}...`);
			else if (t === "timer") process.stdout.write(`${I}  ${l} ${O(g)}`);
			else {
				const z = ".".repeat(Math.floor(w)).slice(0, 3);
				process.stdout.write(`${I}  ${l}${z}`);
			}
			h = h + 1 < n.length ? h + 1 : 0, w = w < n.length ? w + .125 : 0;
		}, r);
	}, N = (m = "", h = 0) => {
		a = !1, clearInterval(c), B();
		const w = h === 0 ? import_picocolors.default.green(C) : h === 1 ? import_picocolors.default.red(L) : import_picocolors.default.red(W);
		l = R(m ?? l), t === "timer" ? process.stdout.write(`${w}  ${l} ${O(g)}
`) : process.stdout.write(`${w}  ${l}
`), E(), s();
	};
	return {
		start: H,
		stop: N,
		message: (m = "") => {
			l = R(m ?? l);
		}
	};
}, Ce = async (t, n) => {
	const r = {}, i = Object.keys(t);
	for (const s of i) {
		const c = t[s], a = await c({ results: r })?.catch((l) => {
			throw l;
		});
		if (typeof n?.onCancel == "function" && pD(a)) {
			r[s] = "canceled", n.onCancel({ results: r });
			continue;
		}
		r[s] = a;
	}
	return r;
}, Te = async (t) => {
	for (const n of t) {
		if (n.enabled === !1) continue;
		const r = Y();
		r.start(n.title);
		const i = await n.task(r.message);
		r.stop(i || n.title);
	}
};

//#endregion
//#region node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/lib/utils.js
/**
* Private utility functions
* @module utils
* @private
*/
var require_utils = /* @__PURE__ */ __commonJSMin(((exports) => {
	var regExpChars = /[|\\{}()[\]^$+*?.]/g;
	var hasOwnProperty = Object.prototype.hasOwnProperty;
	var hasOwn = function(obj, key) {
		return hasOwnProperty.apply(obj, [key]);
	};
	/**
	* Escape characters reserved in regular expressions.
	*
	* If `string` is `undefined` or `null`, the empty string is returned.
	*
	* @param {String} string Input string
	* @return {String} Escaped string
	* @static
	* @private
	*/
	exports.escapeRegExpChars = function(string) {
		// istanbul ignore if
		if (!string) return "";
		return String(string).replace(regExpChars, "\\$&");
	};
	var _ENCODE_HTML_RULES = {
		"&": "&amp;",
		"<": "&lt;",
		">": "&gt;",
		"\"": "&#34;",
		"'": "&#39;"
	};
	var _MATCH_HTML = /[&<>'"]/g;
	function encode_char(c) {
		return _ENCODE_HTML_RULES[c] || c;
	}
	/**
	* Stringified version of constants used by {@link module:utils.escapeXML}.
	*
	* It is used in the process of generating {@link ClientFunction}s.
	*
	* @readonly
	* @type {String}
	*/
	var escapeFuncStr = "var _ENCODE_HTML_RULES = {\n      \"&\": \"&amp;\"\n    , \"<\": \"&lt;\"\n    , \">\": \"&gt;\"\n    , '\"': \"&#34;\"\n    , \"'\": \"&#39;\"\n    }\n  , _MATCH_HTML = /[&<>'\"]/g;\nfunction encode_char(c) {\n  return _ENCODE_HTML_RULES[c] || c;\n};\n";
	/**
	* Escape characters reserved in XML.
	*
	* If `markup` is `undefined` or `null`, the empty string is returned.
	*
	* @implements {EscapeCallback}
	* @param {String} markup Input string
	* @return {String} Escaped string
	* @static
	* @private
	*/
	exports.escapeXML = function(markup) {
		return markup == void 0 ? "" : String(markup).replace(_MATCH_HTML, encode_char);
	};
	function escapeXMLToString() {
		return Function.prototype.toString.call(this) + ";\n" + escapeFuncStr;
	}
	try {
		if (typeof Object.defineProperty === "function") Object.defineProperty(exports.escapeXML, "toString", { value: escapeXMLToString });
		else exports.escapeXML.toString = escapeXMLToString;
	} catch (err) {
		console.warn("Unable to set escapeXML.toString (is the Function prototype frozen?)");
	}
	/**
	* Naive copy of properties from one object to another.
	* Does not recurse into non-scalar properties
	* Does not check to see if the property has a value before copying
	*
	* @param  {Object} to   Destination object
	* @param  {Object} from Source object
	* @return {Object}      Destination object
	* @static
	* @private
	*/
	exports.shallowCopy = function(to, from) {
		from = from || {};
		if (to !== null && to !== void 0) for (var p in from) {
			if (!hasOwn(from, p)) continue;
			if (p === "__proto__" || p === "constructor") continue;
			to[p] = from[p];
		}
		return to;
	};
	/**
	* Naive copy of a list of key names, from one object to another.
	* Only copies property if it is actually defined
	* Does not recurse into non-scalar properties
	*
	* @param  {Object} to   Destination object
	* @param  {Object} from Source object
	* @param  {Array} list List of properties to copy
	* @return {Object}      Destination object
	* @static
	* @private
	*/
	exports.shallowCopyFromList = function(to, from, list) {
		list = list || [];
		from = from || {};
		if (to !== null && to !== void 0) for (var i = 0; i < list.length; i++) {
			var p = list[i];
			if (typeof from[p] != "undefined") {
				if (!hasOwn(from, p)) continue;
				if (p === "__proto__" || p === "constructor") continue;
				to[p] = from[p];
			}
		}
		return to;
	};
	/**
	* Simple in-process cache implementation. Does not implement limits of any
	* sort.
	*
	* @implements {Cache}
	* @static
	* @private
	*/
	exports.cache = {
		_data: {},
		set: function(key, val) {
			this._data[key] = val;
		},
		get: function(key) {
			return this._data[key];
		},
		remove: function(key) {
			delete this._data[key];
		},
		reset: function() {
			this._data = {};
		}
	};
	/**
	* Transforms hyphen case variable into camel case.
	*
	* @param {String} string Hyphen case string
	* @return {String} Camel case string
	* @static
	* @private
	*/
	exports.hyphenToCamel = function(str) {
		return str.replace(/-[a-z]/g, function(match) {
			return match[1].toUpperCase();
		});
	};
	/**
	* Returns a null-prototype object in runtimes that support it
	*
	* @return {Object} Object, prototype will be set to null where possible
	* @static
	* @private
	*/
	exports.createNullProtoObjWherePossible = (function() {
		if (typeof Object.create == "function") return function() {
			return Object.create(null);
		};
		if (!({ __proto__: null } instanceof Object)) return function() {
			return { __proto__: null };
		};
		return function() {
			return {};
		};
	})();
	exports.hasOwnOnlyObject = function(obj) {
		var o = exports.createNullProtoObjWherePossible();
		for (var p in obj) if (hasOwn(obj, p)) o[p] = obj[p];
		return o;
	};
}));

//#endregion
//#region node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/package.json
var require_package = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	module.exports = {
		"name": "ejs",
		"description": "Embedded JavaScript templates",
		"keywords": [
			"template",
			"engine",
			"ejs"
		],
		"version": "3.1.10",
		"author": "Matthew Eernisse <mde@fleegix.org> (http://fleegix.org)",
		"license": "Apache-2.0",
		"bin": { "ejs": "./bin/cli.js" },
		"main": "./lib/ejs.js",
		"jsdelivr": "ejs.min.js",
		"unpkg": "ejs.min.js",
		"repository": {
			"type": "git",
			"url": "git://github.com/mde/ejs.git"
		},
		"bugs": "https://github.com/mde/ejs/issues",
		"homepage": "https://github.com/mde/ejs",
		"dependencies": { "jake": "^10.8.5" },
		"devDependencies": {
			"browserify": "^16.5.1",
			"eslint": "^6.8.0",
			"git-directory-deploy": "^1.5.1",
			"jsdoc": "^4.0.2",
			"lru-cache": "^4.0.1",
			"mocha": "^10.2.0",
			"uglify-js": "^3.3.16"
		},
		"engines": { "node": ">=0.10.0" },
		"scripts": { "test": "npx jake test" }
	};
}));

//#endregion
//#region node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/lib/ejs.js
var require_ejs = /* @__PURE__ */ __commonJSMin(((exports) => {
	/**
	* @file Embedded JavaScript templating engine. {@link http://ejs.co}
	* @author Matthew Eernisse <mde@fleegix.org>
	* @author Tiancheng "Timothy" Gu <timothygu99@gmail.com>
	* @project EJS
	* @license {@link http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0}
	*/
	/**
	* EJS internal functions.
	*
	* Technically this "module" lies in the same file as {@link module:ejs}, for
	* the sake of organization all the private functions re grouped into this
	* module.
	*
	* @module ejs-internal
	* @private
	*/
	/**
	* Embedded JavaScript templating engine.
	*
	* @module ejs
	* @public
	*/
	var fs$1 = __require("fs");
	var path$2 = __require("path");
	var utils = require_utils();
	var scopeOptionWarned = false;
	/** @type {string} */
	var _VERSION_STRING = require_package().version;
	var _DEFAULT_OPEN_DELIMITER = "<";
	var _DEFAULT_CLOSE_DELIMITER = ">";
	var _DEFAULT_DELIMITER = "%";
	var _DEFAULT_LOCALS_NAME = "locals";
	var _NAME = "ejs";
	var _REGEX_STRING = "(<%%|%%>|<%=|<%-|<%_|<%#|<%|%>|-%>|_%>)";
	var _OPTS_PASSABLE_WITH_DATA = [
		"delimiter",
		"scope",
		"context",
		"debug",
		"compileDebug",
		"client",
		"_with",
		"rmWhitespace",
		"strict",
		"filename",
		"async"
	];
	var _OPTS_PASSABLE_WITH_DATA_EXPRESS = _OPTS_PASSABLE_WITH_DATA.concat("cache");
	var _BOM = /^\uFEFF/;
	var _JS_IDENTIFIER = /^[a-zA-Z_$][0-9a-zA-Z_$]*$/;
	/**
	* EJS template function cache. This can be a LRU object from lru-cache NPM
	* module. By default, it is {@link module:utils.cache}, a simple in-process
	* cache that grows continuously.
	*
	* @type {Cache}
	*/
	exports.cache = utils.cache;
	/**
	* Custom file loader. Useful for template preprocessing or restricting access
	* to a certain part of the filesystem.
	*
	* @type {fileLoader}
	*/
	exports.fileLoader = fs$1.readFileSync;
	/**
	* Name of the object containing the locals.
	*
	* This variable is overridden by {@link Options}`.localsName` if it is not
	* `undefined`.
	*
	* @type {String}
	* @public
	*/
	exports.localsName = _DEFAULT_LOCALS_NAME;
	/**
	* Promise implementation -- defaults to the native implementation if available
	* This is mostly just for testability
	*
	* @type {PromiseConstructorLike}
	* @public
	*/
	exports.promiseImpl = new Function("return this;")().Promise;
	/**
	* Get the path to the included file from the parent file path and the
	* specified path.
	*
	* @param {String}  name     specified path
	* @param {String}  filename parent file path
	* @param {Boolean} [isDir=false] whether the parent file path is a directory
	* @return {String}
	*/
	exports.resolveInclude = function(name, filename, isDir) {
		var dirname = path$2.dirname;
		var extname = path$2.extname;
		var resolve = path$2.resolve;
		var includePath = resolve(isDir ? filename : dirname(filename), name);
		if (!extname(name)) includePath += ".ejs";
		return includePath;
	};
	/**
	* Try to resolve file path on multiple directories
	*
	* @param  {String}        name  specified path
	* @param  {Array<String>} paths list of possible parent directory paths
	* @return {String}
	*/
	function resolvePaths(name, paths) {
		var filePath;
		if (paths.some(function(v) {
			filePath = exports.resolveInclude(name, v, true);
			return fs$1.existsSync(filePath);
		})) return filePath;
	}
	/**
	* Get the path to the included file by Options
	*
	* @param  {String}  path    specified path
	* @param  {Options} options compilation options
	* @return {String}
	*/
	function getIncludePath(path, options) {
		var includePath;
		var filePath;
		var views = options.views;
		var match = /^[A-Za-z]+:\\|^\//.exec(path);
		if (match && match.length) {
			path = path.replace(/^\/*/, "");
			if (Array.isArray(options.root)) includePath = resolvePaths(path, options.root);
			else includePath = exports.resolveInclude(path, options.root || "/", true);
		} else {
			if (options.filename) {
				filePath = exports.resolveInclude(path, options.filename);
				if (fs$1.existsSync(filePath)) includePath = filePath;
			}
			if (!includePath && Array.isArray(views)) includePath = resolvePaths(path, views);
			if (!includePath && typeof options.includer !== "function") throw new Error("Could not find the include file \"" + options.escapeFunction(path) + "\"");
		}
		return includePath;
	}
	/**
	* Get the template from a string or a file, either compiled on-the-fly or
	* read from cache (if enabled), and cache the template if needed.
	*
	* If `template` is not set, the file specified in `options.filename` will be
	* read.
	*
	* If `options.cache` is true, this function reads the file from
	* `options.filename` so it must be set prior to calling this function.
	*
	* @memberof module:ejs-internal
	* @param {Options} options   compilation options
	* @param {String} [template] template source
	* @return {(TemplateFunction|ClientFunction)}
	* Depending on the value of `options.client`, either type might be returned.
	* @static
	*/
	function handleCache(options, template) {
		var func;
		var filename = options.filename;
		var hasTemplate = arguments.length > 1;
		if (options.cache) {
			if (!filename) throw new Error("cache option requires a filename");
			func = exports.cache.get(filename);
			if (func) return func;
			if (!hasTemplate) template = fileLoader(filename).toString().replace(_BOM, "");
		} else if (!hasTemplate) {
			// istanbul ignore if: should not happen at all
			if (!filename) throw new Error("Internal EJS error: no file name or template provided");
			template = fileLoader(filename).toString().replace(_BOM, "");
		}
		func = exports.compile(template, options);
		if (options.cache) exports.cache.set(filename, func);
		return func;
	}
	/**
	* Try calling handleCache with the given options and data and call the
	* callback with the result. If an error occurs, call the callback with
	* the error. Used by renderFile().
	*
	* @memberof module:ejs-internal
	* @param {Options} options    compilation options
	* @param {Object} data        template data
	* @param {RenderFileCallback} cb callback
	* @static
	*/
	function tryHandleCache(options, data, cb) {
		var result;
		if (!cb) if (typeof exports.promiseImpl == "function") return new exports.promiseImpl(function(resolve, reject) {
			try {
				result = handleCache(options)(data);
				resolve(result);
			} catch (err) {
				reject(err);
			}
		});
		else throw new Error("Please provide a callback function");
		else {
			try {
				result = handleCache(options)(data);
			} catch (err) {
				return cb(err);
			}
			cb(null, result);
		}
	}
	/**
	* fileLoader is independent
	*
	* @param {String} filePath ejs file path.
	* @return {String} The contents of the specified file.
	* @static
	*/
	function fileLoader(filePath) {
		return exports.fileLoader(filePath);
	}
	/**
	* Get the template function.
	*
	* If `options.cache` is `true`, then the template is cached.
	*
	* @memberof module:ejs-internal
	* @param {String}  path    path for the specified file
	* @param {Options} options compilation options
	* @return {(TemplateFunction|ClientFunction)}
	* Depending on the value of `options.client`, either type might be returned
	* @static
	*/
	function includeFile(path, options) {
		var opts = utils.shallowCopy(utils.createNullProtoObjWherePossible(), options);
		opts.filename = getIncludePath(path, opts);
		if (typeof options.includer === "function") {
			var includerResult = options.includer(path, opts.filename);
			if (includerResult) {
				if (includerResult.filename) opts.filename = includerResult.filename;
				if (includerResult.template) return handleCache(opts, includerResult.template);
			}
		}
		return handleCache(opts);
	}
	/**
	* Re-throw the given `err` in context to the `str` of ejs, `filename`, and
	* `lineno`.
	*
	* @implements {RethrowCallback}
	* @memberof module:ejs-internal
	* @param {Error}  err      Error object
	* @param {String} str      EJS source
	* @param {String} flnm     file name of the EJS file
	* @param {Number} lineno   line number of the error
	* @param {EscapeCallback} esc
	* @static
	*/
	function rethrow(err, str, flnm, lineno, esc) {
		var lines = str.split("\n");
		var start = Math.max(lineno - 3, 0);
		var end = Math.min(lines.length, lineno + 3);
		var filename = esc(flnm);
		var context = lines.slice(start, end).map(function(line, i) {
			var curr = i + start + 1;
			return (curr == lineno ? " >> " : "    ") + curr + "| " + line;
		}).join("\n");
		err.path = filename;
		err.message = (filename || "ejs") + ":" + lineno + "\n" + context + "\n\n" + err.message;
		throw err;
	}
	function stripSemi(str) {
		return str.replace(/;(\s*$)/, "$1");
	}
	/**
	* Compile the given `str` of ejs into a template function.
	*
	* @param {String}  template EJS template
	*
	* @param {Options} [opts] compilation options
	*
	* @return {(TemplateFunction|ClientFunction)}
	* Depending on the value of `opts.client`, either type might be returned.
	* Note that the return type of the function also depends on the value of `opts.async`.
	* @public
	*/
	exports.compile = function compile(template, opts) {
		var templ;
		if (opts && opts.scope) {
			if (!scopeOptionWarned) {
				console.warn("`scope` option is deprecated and will be removed in EJS 3");
				scopeOptionWarned = true;
			}
			if (!opts.context) opts.context = opts.scope;
			delete opts.scope;
		}
		templ = new Template(template, opts);
		return templ.compile();
	};
	/**
	* Render the given `template` of ejs.
	*
	* If you would like to include options but not data, you need to explicitly
	* call this function with `data` being an empty object or `null`.
	*
	* @param {String}   template EJS template
	* @param {Object}  [data={}] template data
	* @param {Options} [opts={}] compilation and rendering options
	* @return {(String|Promise<String>)}
	* Return value type depends on `opts.async`.
	* @public
	*/
	exports.render = function(template, d, o) {
		var data = d || utils.createNullProtoObjWherePossible();
		var opts = o || utils.createNullProtoObjWherePossible();
		if (arguments.length == 2) utils.shallowCopyFromList(opts, data, _OPTS_PASSABLE_WITH_DATA);
		return handleCache(opts, template)(data);
	};
	/**
	* Render an EJS file at the given `path` and callback `cb(err, str)`.
	*
	* If you would like to include options but not data, you need to explicitly
	* call this function with `data` being an empty object or `null`.
	*
	* @param {String}             path     path to the EJS file
	* @param {Object}            [data={}] template data
	* @param {Options}           [opts={}] compilation and rendering options
	* @param {RenderFileCallback} cb callback
	* @public
	*/
	exports.renderFile = function() {
		var args = Array.prototype.slice.call(arguments);
		var filename = args.shift();
		var cb;
		var opts = { filename };
		var data;
		var viewOpts;
		if (typeof arguments[arguments.length - 1] == "function") cb = args.pop();
		if (args.length) {
			data = args.shift();
			if (args.length) utils.shallowCopy(opts, args.pop());
			else {
				if (data.settings) {
					if (data.settings.views) opts.views = data.settings.views;
					if (data.settings["view cache"]) opts.cache = true;
					viewOpts = data.settings["view options"];
					if (viewOpts) utils.shallowCopy(opts, viewOpts);
				}
				utils.shallowCopyFromList(opts, data, _OPTS_PASSABLE_WITH_DATA_EXPRESS);
			}
			opts.filename = filename;
		} else data = utils.createNullProtoObjWherePossible();
		return tryHandleCache(opts, data, cb);
	};
	/**
	* Clear intermediate JavaScript cache. Calls {@link Cache#reset}.
	* @public
	*/
	/**
	* EJS template class
	* @public
	*/
	exports.Template = Template;
	exports.clearCache = function() {
		exports.cache.reset();
	};
	function Template(text, optsParam) {
		var opts = utils.hasOwnOnlyObject(optsParam);
		var options = utils.createNullProtoObjWherePossible();
		this.templateText = text;
		/** @type {string | null} */
		this.mode = null;
		this.truncate = false;
		this.currentLine = 1;
		this.source = "";
		options.client = opts.client || false;
		options.escapeFunction = opts.escape || opts.escapeFunction || utils.escapeXML;
		options.compileDebug = opts.compileDebug !== false;
		options.debug = !!opts.debug;
		options.filename = opts.filename;
		options.openDelimiter = opts.openDelimiter || exports.openDelimiter || _DEFAULT_OPEN_DELIMITER;
		options.closeDelimiter = opts.closeDelimiter || exports.closeDelimiter || _DEFAULT_CLOSE_DELIMITER;
		options.delimiter = opts.delimiter || exports.delimiter || _DEFAULT_DELIMITER;
		options.strict = opts.strict || false;
		options.context = opts.context;
		options.cache = opts.cache || false;
		options.rmWhitespace = opts.rmWhitespace;
		options.root = opts.root;
		options.includer = opts.includer;
		options.outputFunctionName = opts.outputFunctionName;
		options.localsName = opts.localsName || exports.localsName || _DEFAULT_LOCALS_NAME;
		options.views = opts.views;
		options.async = opts.async;
		options.destructuredLocals = opts.destructuredLocals;
		options.legacyInclude = typeof opts.legacyInclude != "undefined" ? !!opts.legacyInclude : true;
		if (options.strict) options._with = false;
		else options._with = typeof opts._with != "undefined" ? opts._with : true;
		this.opts = options;
		this.regex = this.createRegex();
	}
	Template.modes = {
		EVAL: "eval",
		ESCAPED: "escaped",
		RAW: "raw",
		COMMENT: "comment",
		LITERAL: "literal"
	};
	Template.prototype = {
		createRegex: function() {
			var str = _REGEX_STRING;
			var delim = utils.escapeRegExpChars(this.opts.delimiter);
			var open = utils.escapeRegExpChars(this.opts.openDelimiter);
			var close = utils.escapeRegExpChars(this.opts.closeDelimiter);
			str = str.replace(/%/g, delim).replace(/</g, open).replace(/>/g, close);
			return new RegExp(str);
		},
		compile: function() {
			/** @type {string} */
			var src;
			/** @type {ClientFunction} */
			var fn;
			var opts = this.opts;
			var prepended = "";
			var appended = "";
			/** @type {EscapeCallback} */
			var escapeFn = opts.escapeFunction;
			/** @type {FunctionConstructor} */
			var ctor;
			/** @type {string} */
			var sanitizedFilename = opts.filename ? JSON.stringify(opts.filename) : "undefined";
			if (!this.source) {
				this.generateSource();
				prepended += "  var __output = \"\";\n  function __append(s) { if (s !== undefined && s !== null) __output += s }\n";
				if (opts.outputFunctionName) {
					if (!_JS_IDENTIFIER.test(opts.outputFunctionName)) throw new Error("outputFunctionName is not a valid JS identifier.");
					prepended += "  var " + opts.outputFunctionName + " = __append;\n";
				}
				if (opts.localsName && !_JS_IDENTIFIER.test(opts.localsName)) throw new Error("localsName is not a valid JS identifier.");
				if (opts.destructuredLocals && opts.destructuredLocals.length) {
					var destructuring = "  var __locals = (" + opts.localsName + " || {}),\n";
					for (var i = 0; i < opts.destructuredLocals.length; i++) {
						var name = opts.destructuredLocals[i];
						if (!_JS_IDENTIFIER.test(name)) throw new Error("destructuredLocals[" + i + "] is not a valid JS identifier.");
						if (i > 0) destructuring += ",\n  ";
						destructuring += name + " = __locals." + name;
					}
					prepended += destructuring + ";\n";
				}
				if (opts._with !== false) {
					prepended += "  with (" + opts.localsName + " || {}) {\n";
					appended += "  }\n";
				}
				appended += "  return __output;\n";
				this.source = prepended + this.source + appended;
			}
			if (opts.compileDebug) src = "var __line = 1\n  , __lines = " + JSON.stringify(this.templateText) + "\n  , __filename = " + sanitizedFilename + ";\ntry {\n" + this.source + "} catch (e) {\n  rethrow(e, __lines, __filename, __line, escapeFn);\n}\n";
			else src = this.source;
			if (opts.client) {
				src = "escapeFn = escapeFn || " + escapeFn.toString() + ";\n" + src;
				if (opts.compileDebug) src = "rethrow = rethrow || " + rethrow.toString() + ";\n" + src;
			}
			if (opts.strict) src = "\"use strict\";\n" + src;
			if (opts.debug) console.log(src);
			if (opts.compileDebug && opts.filename) src = src + "\n//# sourceURL=" + sanitizedFilename + "\n";
			try {
				if (opts.async) try {
					ctor = new Function("return (async function(){}).constructor;")();
				} catch (e) {
					if (e instanceof SyntaxError) throw new Error("This environment does not support async/await");
					else throw e;
				}
				else ctor = Function;
				fn = new ctor(opts.localsName + ", escapeFn, include, rethrow", src);
			} catch (e) {
				// istanbul ignore else
				if (e instanceof SyntaxError) {
					if (opts.filename) e.message += " in " + opts.filename;
					e.message += " while compiling ejs\n\n";
					e.message += "If the above error is not helpful, you may want to try EJS-Lint:\n";
					e.message += "https://github.com/RyanZim/EJS-Lint";
					if (!opts.async) {
						e.message += "\n";
						e.message += "Or, if you meant to create an async function, pass `async: true` as an option.";
					}
				}
				throw e;
			}
			var returnedFn = opts.client ? fn : function anonymous(data) {
				var include = function(path, includeData) {
					var d = utils.shallowCopy(utils.createNullProtoObjWherePossible(), data);
					if (includeData) d = utils.shallowCopy(d, includeData);
					return includeFile(path, opts)(d);
				};
				return fn.apply(opts.context, [
					data || utils.createNullProtoObjWherePossible(),
					escapeFn,
					include,
					rethrow
				]);
			};
			if (opts.filename && typeof Object.defineProperty === "function") {
				var filename = opts.filename;
				var basename = path$2.basename(filename, path$2.extname(filename));
				try {
					Object.defineProperty(returnedFn, "name", {
						value: basename,
						writable: false,
						enumerable: false,
						configurable: true
					});
				} catch (e) {}
			}
			return returnedFn;
		},
		generateSource: function() {
			if (this.opts.rmWhitespace) this.templateText = this.templateText.replace(/[\r\n]+/g, "\n").replace(/^\s+|\s+$/gm, "");
			this.templateText = this.templateText.replace(/[ \t]*<%_/gm, "<%_").replace(/_%>[ \t]*/gm, "_%>");
			var self = this;
			var matches = this.parseTemplateText();
			var d = this.opts.delimiter;
			var o = this.opts.openDelimiter;
			var c = this.opts.closeDelimiter;
			if (matches && matches.length) matches.forEach(function(line, index) {
				var closing;
				if (line.indexOf(o + d) === 0 && line.indexOf(o + d + d) !== 0) {
					closing = matches[index + 2];
					if (!(closing == d + c || closing == "-" + d + c || closing == "_" + d + c)) throw new Error("Could not find matching close tag for \"" + line + "\".");
				}
				self.scanLine(line);
			});
		},
		parseTemplateText: function() {
			var str = this.templateText;
			var pat = this.regex;
			var result = pat.exec(str);
			var arr = [];
			var firstPos;
			while (result) {
				firstPos = result.index;
				if (firstPos !== 0) {
					arr.push(str.substring(0, firstPos));
					str = str.slice(firstPos);
				}
				arr.push(result[0]);
				str = str.slice(result[0].length);
				result = pat.exec(str);
			}
			if (str) arr.push(str);
			return arr;
		},
		_addOutput: function(line) {
			if (this.truncate) {
				line = line.replace(/^(?:\r\n|\r|\n)/, "");
				this.truncate = false;
			}
			if (!line) return line;
			line = line.replace(/\\/g, "\\\\");
			line = line.replace(/\n/g, "\\n");
			line = line.replace(/\r/g, "\\r");
			line = line.replace(/"/g, "\\\"");
			this.source += "    ; __append(\"" + line + "\")\n";
		},
		scanLine: function(line) {
			var self = this;
			var d = this.opts.delimiter;
			var o = this.opts.openDelimiter;
			var c = this.opts.closeDelimiter;
			var newLineCount = 0;
			newLineCount = line.split("\n").length - 1;
			switch (line) {
				case o + d:
				case o + d + "_":
					this.mode = Template.modes.EVAL;
					break;
				case o + d + "=":
					this.mode = Template.modes.ESCAPED;
					break;
				case o + d + "-":
					this.mode = Template.modes.RAW;
					break;
				case o + d + "#":
					this.mode = Template.modes.COMMENT;
					break;
				case o + d + d:
					this.mode = Template.modes.LITERAL;
					this.source += "    ; __append(\"" + line.replace(o + d + d, o + d) + "\")\n";
					break;
				case d + d + c:
					this.mode = Template.modes.LITERAL;
					this.source += "    ; __append(\"" + line.replace(d + d + c, d + c) + "\")\n";
					break;
				case d + c:
				case "-" + d + c:
				case "_" + d + c:
					if (this.mode == Template.modes.LITERAL) this._addOutput(line);
					this.mode = null;
					this.truncate = line.indexOf("-") === 0 || line.indexOf("_") === 0;
					break;
				default: if (this.mode) {
					switch (this.mode) {
						case Template.modes.EVAL:
						case Template.modes.ESCAPED:
						case Template.modes.RAW: if (line.lastIndexOf("//") > line.lastIndexOf("\n")) line += "\n";
					}
					switch (this.mode) {
						case Template.modes.EVAL:
							this.source += "    ; " + line + "\n";
							break;
						case Template.modes.ESCAPED:
							this.source += "    ; __append(escapeFn(" + stripSemi(line) + "))\n";
							break;
						case Template.modes.RAW:
							this.source += "    ; __append(" + stripSemi(line) + ")\n";
							break;
						case Template.modes.COMMENT: break;
						case Template.modes.LITERAL:
							this._addOutput(line);
							break;
					}
				} else this._addOutput(line);
			}
			if (self.opts.compileDebug && newLineCount) {
				this.currentLine += newLineCount;
				this.source += "    ; __line = " + this.currentLine + "\n";
			}
		}
	};
	/**
	* Escape characters reserved in XML.
	*
	* This is simply an export of {@link module:utils.escapeXML}.
	*
	* If `markup` is `undefined` or `null`, the empty string is returned.
	*
	* @param {String} markup Input string
	* @return {String} Escaped string
	* @public
	* @func
	* */
	exports.escapeXML = utils.escapeXML;
	/**
	* Express.js support.
	*
	* This is an alias for {@link module:ejs.renderFile}, in order to support
	* Express.js out-of-the-box.
	*
	* @func
	*/
	exports.__express = exports.renderFile;
	/**
	* Version of EJS.
	*
	* @readonly
	* @type {String}
	* @public
	*/
	exports.VERSION = _VERSION_STRING;
	/**
	* Name for detection of EJS.
	*
	* @readonly
	* @type {String}
	* @public
	*/
	exports.name = _NAME;
	/* istanbul ignore if */
	if (typeof window != "undefined") window.ejs = exports;
}));

//#endregion
//#region utils/banners.ts
var import_ejs = /* @__PURE__ */ __toESM(require_ejs(), 1);
const defaultBanner = "Vue.js - The Progressive JavaScript Framework";
const gradientBanner = "\x1B[38;2;66;211;146mV\x1B[39m\x1B[38;2;66;211;146mu\x1B[39m\x1B[38;2;66;211;146me\x1B[39m\x1B[38;2;66;211;146m.\x1B[39m\x1B[38;2;66;211;146mj\x1B[39m\x1B[38;2;67;209;149ms\x1B[39m \x1B[38;2;68;206;152m-\x1B[39m \x1B[38;2;69;204;155mT\x1B[39m\x1B[38;2;70;201;158mh\x1B[39m\x1B[38;2;71;199;162me\x1B[39m \x1B[38;2;72;196;165mP\x1B[39m\x1B[38;2;73;194;168mr\x1B[39m\x1B[38;2;74;192;171mo\x1B[39m\x1B[38;2;75;189;174mg\x1B[39m\x1B[38;2;76;187;177mr\x1B[39m\x1B[38;2;77;184;180me\x1B[39m\x1B[38;2;78;182;183ms\x1B[39m\x1B[38;2;79;179;186ms\x1B[39m\x1B[38;2;80;177;190mi\x1B[39m\x1B[38;2;81;175;193mv\x1B[39m\x1B[38;2;82;172;196me\x1B[39m \x1B[38;2;83;170;199mJ\x1B[39m\x1B[38;2;83;167;202ma\x1B[39m\x1B[38;2;84;165;205mv\x1B[39m\x1B[38;2;85;162;208ma\x1B[39m\x1B[38;2;86;160;211mS\x1B[39m\x1B[38;2;87;158;215mc\x1B[39m\x1B[38;2;88;155;218mr\x1B[39m\x1B[38;2;89;153;221mi\x1B[39m\x1B[38;2;90;150;224mp\x1B[39m\x1B[38;2;91;148;227mt\x1B[39m \x1B[38;2;92;145;230mF\x1B[39m\x1B[38;2;93;143;233mr\x1B[39m\x1B[38;2;94;141;236ma\x1B[39m\x1B[38;2;95;138;239mm\x1B[39m\x1B[38;2;96;136;243me\x1B[39m\x1B[38;2;97;133;246mw\x1B[39m\x1B[38;2;98;131;249mo\x1B[39m\x1B[38;2;99;128;252mr\x1B[39m\x1B[38;2;100;126;255mk\x1B[39m";

//#endregion
//#region utils/deepMerge.ts
const isObject = (val) => val && typeof val === "object";
const mergeArrayWithDedupe = (a, b) => Array.from(new Set([...a, ...b]));
/**
* Recursively merge the content of the new object to the existing one
* @param {Object} target the existing object
* @param {Object} obj the new object
*/
function deepMerge(target, obj) {
	for (const key of Object.keys(obj)) {
		const oldVal = target[key];
		const newVal = obj[key];
		if (Array.isArray(oldVal) && Array.isArray(newVal)) target[key] = mergeArrayWithDedupe(oldVal, newVal);
		else if (isObject(oldVal) && isObject(newVal)) target[key] = deepMerge(oldVal, newVal);
		else target[key] = newVal;
	}
	return target;
}

//#endregion
//#region utils/sortDependencies.ts
function sortDependencies(packageJson) {
	const sorted = {};
	for (const depType of [
		"dependencies",
		"devDependencies",
		"peerDependencies",
		"optionalDependencies"
	]) if (packageJson[depType]) {
		sorted[depType] = {};
		Object.keys(packageJson[depType]).sort().forEach((name) => {
			sorted[depType][name] = packageJson[depType][name];
		});
	}
	return {
		...packageJson,
		...sorted
	};
}

//#endregion
//#region utils/renderTemplate.ts
/**
* Renders a template folder/file to the file system,
* by recursively copying all files under the `src` directory,
* with the following exception:
*   - `_filename` should be renamed to `.filename`
*   - Fields in `package.json` should be recursively merged
* @param {string} src source filename to copy
* @param {string} dest destination filename of the copy operation
*/
function renderTemplate(src, dest, callbacks) {
	if (fs.statSync(src).isDirectory()) {
		if (path$1.basename(src) === "node_modules") return;
		fs.mkdirSync(dest, { recursive: true });
		for (const file of fs.readdirSync(src)) renderTemplate(path$1.resolve(src, file), path$1.resolve(dest, file), callbacks);
		return;
	}
	const filename = path$1.basename(src);
	if (filename === "package.json" && fs.existsSync(dest)) {
		const pkg = sortDependencies(deepMerge(JSON.parse(fs.readFileSync(dest, "utf8")), JSON.parse(fs.readFileSync(src, "utf8"))));
		fs.writeFileSync(dest, JSON.stringify(pkg, null, 2) + "\n");
		return;
	}
	if (filename === "extensions.json" && fs.existsSync(dest)) {
		const extensions = deepMerge(JSON.parse(fs.readFileSync(dest, "utf8")), JSON.parse(fs.readFileSync(src, "utf8")));
		fs.writeFileSync(dest, JSON.stringify(extensions, null, 2) + "\n");
		return;
	}
	if (filename === "settings.json" && fs.existsSync(dest)) {
		const settings = deepMerge(JSON.parse(fs.readFileSync(dest, "utf8")), JSON.parse(fs.readFileSync(src, "utf8")));
		fs.writeFileSync(dest, JSON.stringify(settings, null, 2) + "\n");
		return;
	}
	if (filename.startsWith("_")) dest = path$1.resolve(path$1.dirname(dest), filename.replace(/^_/, "."));
	if (filename === "_gitignore" && fs.existsSync(dest)) {
		const existing = fs.readFileSync(dest, "utf8");
		const newGitignore = fs.readFileSync(src, "utf8");
		fs.writeFileSync(dest, existing + "\n" + newGitignore);
		return;
	}
	if (filename.endsWith(".data.mjs")) {
		dest = dest.replace(/\.data\.mjs$/, "");
		callbacks.push(async (dataStore) => {
			const getData = (await import(pathToFileURL(src).toString())).default;
			dataStore[dest] = await getData({ oldData: dataStore[dest] || {} });
		});
		return;
	}
	fs.copyFileSync(src, dest);
}

//#endregion
//#region utils/directoryTraverse.ts
function preOrderDirectoryTraverse(dir, dirCallback, fileCallback) {
	for (const filename of fs.readdirSync(dir)) {
		if (filename === ".git") continue;
		const fullpath = path$1.resolve(dir, filename);
		if (fs.lstatSync(fullpath).isDirectory()) {
			dirCallback(fullpath);
			if (fs.existsSync(fullpath)) preOrderDirectoryTraverse(fullpath, dirCallback, fileCallback);
			continue;
		}
		fileCallback(fullpath);
	}
}
const dotGitDirectoryState = { hasDotGitDirectory: false };
function postOrderDirectoryTraverse(dir, dirCallback, fileCallback) {
	for (const filename of fs.readdirSync(dir)) {
		if (filename === ".git") {
			dotGitDirectoryState.hasDotGitDirectory = true;
			continue;
		}
		const fullpath = path$1.resolve(dir, filename);
		if (fs.lstatSync(fullpath).isDirectory()) {
			postOrderDirectoryTraverse(fullpath, dirCallback, fileCallback);
			dirCallback(fullpath);
			continue;
		}
		fileCallback(fullpath);
	}
}

//#endregion
//#region utils/getCommand.ts
function getCommand(packageManager, scriptName, args) {
	if (scriptName === "install") return packageManager === "yarn" ? "yarn" : `${packageManager} install`;
	if (scriptName === "build") return packageManager === "npm" || packageManager === "bun" ? `${packageManager} run build` : `${packageManager} build`;
	if (args) return packageManager === "npm" ? `npm run ${scriptName} -- ${args}` : `${packageManager} ${scriptName} ${args}`;
	else return packageManager === "npm" ? `npm run ${scriptName}` : `${packageManager} ${scriptName}`;
}

//#endregion
//#region utils/generateReadme.ts
const sfcTypeSupportDoc = [
	"",
	"## Type Support for `.vue` Imports in TS",
	"",
	"TypeScript cannot handle type information for `.vue` imports by default, so we replace the `tsc` CLI with `vue-tsc` for type checking. In editors, we need [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) to make the TypeScript language service aware of `.vue` types.",
	""
].join("\n");
function generateReadme({ projectName, packageManager, needsTypeScript, needsCypress, needsNightwatch, needsCypressCT, needsNightwatchCT, needsPlaywright, needsVitest, needsEslint }) {
	const commandFor = (scriptName, args) => getCommand(packageManager, scriptName, args);
	let readme = `# ${projectName}

This template should help get you started developing with Vue 3 in Vite.

## Recommended IDE Setup

[VS Code](https://code.visualstudio.com/) + [Vue (Official)](https://marketplace.visualstudio.com/items?itemName=Vue.volar) (and disable Vetur).

## Recommended Browser Setup

- Chromium-based browsers (Chrome, Edge, Brave, etc.):
  - [Vue.js devtools](https://chromewebstore.google.com/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd)
  - [Turn on Custom Object Formatter in Chrome DevTools](http://bit.ly/object-formatters)
- Firefox:
  - [Vue.js devtools](https://addons.mozilla.org/en-US/firefox/addon/vue-js-devtools/)
  - [Turn on Custom Object Formatter in Firefox DevTools](https://fxdx.dev/firefox-devtools-custom-object-formatters/)
${needsTypeScript ? sfcTypeSupportDoc : ""}
## Customize configuration

See [Vite Configuration Reference](https://vite.dev/config/).

## Project Setup

`;
	let installCommand = commandFor("install");
	if (packageManager === "pnpm" && needsNightwatch) installCommand += `\npnpm approve-builds # for pnpm 10+`;
	let npmScriptsDescriptions = `\`\`\`sh
${installCommand}
\`\`\`

### Compile and Hot-Reload for Development

\`\`\`sh
${commandFor("dev")}
\`\`\`

### ${needsTypeScript ? "Type-Check, " : ""}Compile and Minify for Production

\`\`\`sh
${commandFor("build")}
\`\`\`
`;
	if (needsVitest) npmScriptsDescriptions += `
### Run Unit Tests with [Vitest](https://vitest.dev/)

\`\`\`sh
${commandFor("test:unit")}
\`\`\`
`;
	if (needsCypressCT) npmScriptsDescriptions += `
### Run Headed Component Tests with [Cypress Component Testing](https://on.cypress.io/component)

\`\`\`sh
${commandFor("test:unit:dev")} # or \`${commandFor("test:unit")}\` for headless testing
\`\`\`
`;
	if (needsCypress) npmScriptsDescriptions += `
### Run End-to-End Tests with [Cypress](https://www.cypress.io/)

\`\`\`sh
${commandFor("test:e2e:dev")}
\`\`\`

This runs the end-to-end tests against the Vite development server.
It is much faster than the production build.

But it's still recommended to test the production build with \`test:e2e\` before deploying (e.g. in CI environments):

\`\`\`sh
${commandFor("build")}
${commandFor("test:e2e")}
\`\`\`
`;
	if (needsNightwatch) npmScriptsDescriptions += `
### Run End-to-End Tests with [Nightwatch](https://nightwatchjs.org/)

\`\`\`sh
# When using CI, the project must be built first.
${commandFor("build")}

# Runs the end-to-end tests
${commandFor("test:e2e")}
# Runs the tests only on Chrome
${commandFor("test:e2e", "--env chrome")}
# Runs the tests of a specific file
${commandFor("test:e2e", `tests/e2e/example.${needsTypeScript ? "ts" : "js"}`)}
# Runs the tests in debug mode
${commandFor("test:e2e", "--debug")}
\`\`\`
    `;
	if (needsNightwatchCT) npmScriptsDescriptions += `
### Run Headed Component Tests with [Nightwatch Component Testing](https://nightwatchjs.org/guide/component-testing/introduction.html)
  
\`\`\`sh
${commandFor("test:unit")}
${commandFor("test:unit -- --headless # for headless testing")}
\`\`\`
`;
	if (needsPlaywright) npmScriptsDescriptions += `
### Run End-to-End Tests with [Playwright](https://playwright.dev)

\`\`\`sh
# Install browsers for the first run
npx playwright install

# When testing on CI, must build the project first
${commandFor("build")}

# Runs the end-to-end tests
${commandFor("test:e2e")}
# Runs the tests only on Chromium
${commandFor("test:e2e", "--project=chromium")}
# Runs the tests of a specific file
${commandFor("test:e2e", "tests/example.spec.ts")}
# Runs the tests in debug mode
${commandFor("test:e2e", "--debug")}
\`\`\`
`;
	if (needsEslint) npmScriptsDescriptions += `
### Lint with [ESLint](https://eslint.org/)

\`\`\`sh
${commandFor("lint")}
\`\`\`
`;
	readme += npmScriptsDescriptions;
	return readme;
}

//#endregion
//#region utils/getLanguage.ts
/**
*
* This function is used to link obtained locale with correct locale file in order to make locales reusable
*
* @param locale the obtained locale
* @returns locale that linked with correct name
*/
function linkLocale(locale) {
	if (locale === "C") return "en-US";
	let linkedLocale;
	try {
		linkedLocale = Intl.getCanonicalLocales(locale)[0];
	} catch (error) {
		console.log(`${error.toString()}, invalid language tag: "${locale}"\n`);
	}
	switch (linkedLocale) {
		case "zh-TW":
		case "zh-HK":
		case "zh-MO":
			linkedLocale = "zh-Hant";
			break;
		case "zh-CN":
		case "zh-SG":
			linkedLocale = "zh-Hans";
			break;
		default: linkedLocale = locale;
	}
	return linkedLocale;
}
function getLocale() {
	return linkLocale((process.env.LC_ALL || process.env.LC_MESSAGES || process.env.LANG || Intl.DateTimeFormat().resolvedOptions().locale || "en-US").split(".")[0].replace("_", "-"));
}
async function loadLanguageFile(filePath) {
	return await fs.promises.readFile(filePath, "utf-8").then((data) => {
		const parsedData = JSON.parse(data);
		if (parsedData) return parsedData;
	});
}
async function getLanguage(localesRoot) {
	const locale = getLocale();
	const languageFilePath = path$1.resolve(localesRoot, `${locale}.json`);
	const fallbackPath = path$1.resolve(localesRoot, "en-US.json");
	return fs.existsSync(languageFilePath) ? await loadLanguageFile(languageFilePath) : await loadLanguageFile(fallbackPath);
}

//#endregion
//#region utils/trimBoilerplate.ts
function replaceContent(filepath, replacer) {
	const content = fs.readFileSync(filepath, "utf8");
	fs.writeFileSync(filepath, replacer(content));
}
function trimBoilerplate(rootDir) {
	const srcDir = path.resolve(rootDir, "src");
	for (const filename of fs.readdirSync(srcDir)) {
		if ([
			"main.js",
			"main.ts",
			"router",
			"stores"
		].includes(filename)) continue;
		const fullpath = path.resolve(srcDir, filename);
		fs.rmSync(fullpath, { recursive: true });
	}
}
function removeCSSImport(rootDir, needsTypeScript, needsCypressCT) {
	replaceContent(path.resolve(rootDir, needsTypeScript ? "src/main.ts" : "src/main.js"), (content) => content.replace("import './assets/main.css'\n\n", ""));
	if (needsCypressCT) replaceContent(path.resolve(rootDir, needsTypeScript ? "cypress/support/component.ts" : "cypress/support/component.js"), (content) => content.replace("import '@/assets/main.css'", "// import '@/assets/main.css'"));
}
function emptyRouterConfig(rootDir, needsTypeScript) {
	const srcDir = path.resolve(rootDir, "src");
	replaceContent(path.resolve(srcDir, needsTypeScript ? "router/index.ts" : "router/index.js"), (content) => content.replace(`import HomeView from '../views/HomeView.vue'\n`, "").replace(/routes:\s*\[[\s\S]*?\],/, "routes: [],"));
}

//#endregion
//#region utils/applyVueBeta.ts
const CORE_VUE_PACKAGES = [
	"vue",
	"@vue/compiler-core",
	"@vue/compiler-dom",
	"@vue/compiler-sfc",
	"@vue/compiler-ssr",
	"@vue/compiler-vapor",
	"@vue/reactivity",
	"@vue/runtime-core",
	"@vue/runtime-dom",
	"@vue/runtime-vapor",
	"@vue/server-renderer",
	"@vue/shared",
	"@vue/compat"
];
function generateOverridesMap() {
	return Object.fromEntries(CORE_VUE_PACKAGES.map((name) => [name, "beta"]));
}
/**
* Apply Vue 3.6 beta overrides to the project based on the package manager.
* Different package managers have different mechanisms for version overrides:
* - npm/bun: uses `overrides` field in package.json
* - yarn: uses `resolutions` field in package.json
* - pnpm: uses `overrides` and `peerDependencyRules` in pnpm-workspace.yaml
*/
function applyVueBeta(root, packageManager, pkg) {
	const overrides = generateOverridesMap();
	if (packageManager === "npm" || packageManager === "bun") {
		pkg.overrides = {
			...pkg.overrides,
			...overrides
		};
		for (const dependencyName of CORE_VUE_PACKAGES) for (const dependencyType of [
			"dependencies",
			"devDependencies",
			"optionalDependencies"
		]) if (pkg[dependencyType]?.[dependencyName]) pkg[dependencyType][dependencyName] = overrides[dependencyName];
	} else if (packageManager === "yarn") pkg.resolutions = {
		...pkg.resolutions,
		...overrides
	};
	else if (packageManager === "pnpm") {
		const yamlContent = `overrides:
${Object.entries(overrides).map(([key, value]) => `  '${key}': '${value}'`).join("\n")}

peerDependencyRules:
  allowAny:
    - 'vue'
`;
		fs.writeFileSync(path$1.resolve(root, "pnpm-workspace.yaml"), yamlContent, "utf-8");
	}
}

//#endregion
//#region utils/packageManager.ts
/**
* Infers the package manager from the user agent string.
* Falls back to npm if unable to detect.
*/
function inferPackageManager() {
	const userAgent = process.env.npm_config_user_agent ?? "";
	if (/pnpm/.test(userAgent)) return "pnpm";
	if (/yarn/.test(userAgent)) return "yarn";
	if (/bun/.test(userAgent)) return "bun";
	return "npm";
}
/**
* Creates an ordered list of package managers with the preferred one first.
*/
function getPackageManagerOptions(preferred) {
	return [preferred, ...[
		"npm",
		"pnpm",
		"yarn",
		"bun"
	].filter((pm) => pm !== preferred)];
}

//#endregion
//#region package.json
var name = "create-vue";
var version = "3.21.1";

//#endregion
//#region index.ts
const language = await getLanguage(fileURLToPath(new URL("./locales", import.meta.url)));
const FEATURE_FLAGS = [
	"default",
	"ts",
	"typescript",
	"jsx",
	"router",
	"vue-router",
	"pinia",
	"tests",
	"with-tests",
	"vitest",
	"cypress",
	"nightwatch",
	"playwright",
	"eslint",
	"prettier",
	"eslint-with-prettier",
	"oxlint",
	"oxfmt",
	"vite-beta",
	"vue-beta"
];
const FEATURE_OPTIONS = [
	{
		value: "typescript",
		label: language.needsTypeScript.message
	},
	{
		value: "jsx",
		label: language.needsJsx.message
	},
	{
		value: "router",
		label: language.needsRouter.message
	},
	{
		value: "pinia",
		label: language.needsPinia.message
	},
	{
		value: "vitest",
		label: language.needsVitest.message
	},
	{
		value: "e2e",
		label: language.needsE2eTesting.message
	},
	{
		value: "eslint",
		label: language.needsEslint.message
	},
	{
		value: "prettier",
		label: language.needsPrettier.message
	}
];
const EXPERIMENTAL_FEATURE_OPTIONS = [
	{
		value: "oxfmt",
		label: language.needsOxfmt.message
	},
	{
		value: "vite-beta",
		label: language.needsViteBeta.message
	},
	{
		value: "vue-beta",
		label: language.needsVueBeta.message
	}
];
function isValidPackageName(projectName) {
	return /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/.test(projectName);
}
function toValidPackageName(projectName) {
	return projectName.trim().toLowerCase().replace(/\s+/g, "-").replace(/^[._]/, "").replace(/[^a-z0-9-~]+/g, "-");
}
function canSkipEmptying(dir) {
	if (!fs.existsSync(dir)) return true;
	const files = fs.readdirSync(dir);
	if (files.length === 0) return true;
	if (files.length === 1 && files[0] === ".git") {
		dotGitDirectoryState.hasDotGitDirectory = true;
		return true;
	}
	return false;
}
function emptyDir(dir) {
	if (!fs.existsSync(dir)) return;
	postOrderDirectoryTraverse(dir, (dir) => fs.rmdirSync(dir), (file) => fs.unlinkSync(file));
}
async function unwrapPrompt(maybeCancelPromise) {
	const result = await maybeCancelPromise;
	if (pD(result)) {
		xe((0, import_picocolors.red)("✖") + ` ${language.errors.operationCancelled}`);
		process.exit(0);
	}
	return result;
}
const helpMessage = `\
Usage: create-vue [FEATURE_FLAGS...] [OPTIONS...] [DIRECTORY]

Create a new Vue.js project.
Runs in interactive mode if started without feature flags, or if DIRECTORY is missing or not a valid package name.

Options:
  --force
    Create the project even if the directory is not empty.
  --bare
    Create a barebone project without example code.
  --help
    Display this help message.
  --version
    Display the version number of this CLI.

Available feature flags:
  --default
    Create a project with the default configuration without any additional features.
  --ts, --typescript
    Add TypeScript support.
  --jsx
    Add JSX support.
  --router, --vue-router
    Add Vue Router for SPA development.
  --pinia
    Add Pinia for state management.
  --vitest
    Add Vitest for unit testing.
  --cypress
    Add Cypress for end-to-end testing.
    If used without ${(0, import_picocolors.cyan)("--vitest")}, it will also add Cypress Component Testing.
  --playwright
    Add Playwright for end-to-end testing.
  --nightwatch
    Add Nightwatch for end-to-end testing.
    If used without ${(0, import_picocolors.cyan)("--vitest")}, it will also add Nightwatch Component Testing.
  --eslint
    Add ESLint for code quality.
  --prettier
    Add Prettier for code formatting.
  --oxfmt
    Add Oxfmt for code formatting.
  --vite-beta
    Use Vite 8 Beta instead of Vite for building the project.
  --vue-beta
    Use Vue 3.6 Beta. Requires specifying a package manager in interactive mode.

Unstable feature flags:
  --tests, --with-tests
    Add both unit testing and end-to-end testing support.
    Currently equivalent to ${(0, import_picocolors.cyan)("--vitest --cypress")}, but may change in the future.

Deprecated feature flags:
  --eslint-with-prettier
    Please use ${(0, import_picocolors.cyan)("--eslint --prettier")} instead.
  --oxlint
    Oxlint is now always included when ESLint is selected.
  --rolldown-vite
    Please use ${(0, import_picocolors.cyan)("--vite-beta")} instead.
`;
async function init() {
	const cwd = process.cwd();
	const args = process.argv.slice(2);
	const flags = [
		...FEATURE_FLAGS,
		"force",
		"bare",
		"help",
		"version"
	];
	const { values: argv, positionals } = parseArgs({
		args,
		options: Object.fromEntries(flags.map((key) => [key, { type: "boolean" }])),
		strict: true,
		allowPositionals: true
	});
	if (argv.help) {
		console.log(helpMessage);
		process.exit(0);
	}
	if (argv.version) {
		console.log(`${name} v${version}`);
		process.exit(0);
	}
	const isFeatureFlagsUsed = FEATURE_FLAGS.some((flag) => typeof argv[flag] === "boolean");
	let targetDir = positionals[0];
	const defaultProjectName = targetDir || "vue-project";
	const forceOverwrite = argv.force;
	const inferredPackageManager = inferPackageManager();
	const result = {
		projectName: defaultProjectName,
		shouldOverwrite: forceOverwrite,
		packageName: defaultProjectName,
		features: [],
		e2eFramework: void 0,
		experimentFeatures: [],
		needsBareboneTemplates: false
	};
	Ie(process.stdout.isTTY && process.stdout.getColorDepth() > 8 ? gradientBanner : defaultBanner);
	if (!targetDir) targetDir = result.projectName = result.packageName = (await unwrapPrompt(he({
		message: language.projectName.message,
		placeholder: defaultProjectName,
		defaultValue: defaultProjectName,
		validate: (value) => value.length === 0 || value.trim().length > 0 ? void 0 : language.projectName.invalidMessage
	}))).trim();
	if (!canSkipEmptying(targetDir) && !forceOverwrite) {
		result.shouldOverwrite = await unwrapPrompt(ye({
			message: `${targetDir === "." ? language.shouldOverwrite.dirForPrompts.current : `${language.shouldOverwrite.dirForPrompts.target} "${targetDir}"`} ${language.shouldOverwrite.message}`,
			initialValue: false
		}));
		if (!result.shouldOverwrite) {
			xe((0, import_picocolors.red)("✖") + ` ${language.errors.operationCancelled}`);
			process.exit(0);
		}
	}
	if (!isValidPackageName(targetDir)) result.packageName = await unwrapPrompt(he({
		message: language.packageName.message,
		initialValue: toValidPackageName(targetDir),
		validate: (value) => isValidPackageName(value) ? void 0 : language.packageName.invalidMessage
	}));
	if (!isFeatureFlagsUsed) {
		result.features = await unwrapPrompt(fe({
			message: `${language.featureSelection.message} ${(0, import_picocolors.dim)(language.featureSelection.hint)}`,
			options: FEATURE_OPTIONS,
			required: false
		}));
		if (result.features.includes("e2e")) {
			const hasVitest = result.features.includes("vitest");
			result.e2eFramework = await unwrapPrompt(ve({
				message: `${language.e2eSelection.message} ${(0, import_picocolors.dim)(language.e2eSelection.hint)}`,
				options: [
					{
						value: "playwright",
						label: language.e2eSelection.selectOptions.playwright.title,
						hint: language.e2eSelection.selectOptions.playwright.desc
					},
					{
						value: "cypress",
						label: language.e2eSelection.selectOptions.cypress.title,
						hint: hasVitest ? language.e2eSelection.selectOptions.cypress.desc : language.e2eSelection.selectOptions.cypress.hintOnComponentTesting
					},
					{
						value: "nightwatch",
						label: language.e2eSelection.selectOptions.nightwatch.title,
						hint: hasVitest ? language.e2eSelection.selectOptions.nightwatch.desc : language.e2eSelection.selectOptions.nightwatch.hintOnComponentTesting
					}
				]
			}));
		}
		result.experimentFeatures = await unwrapPrompt(fe({
			message: `${language.needsExperimentalFeatures.message} ${(0, import_picocolors.dim)(language.needsExperimentalFeatures.hint)}`,
			options: EXPERIMENTAL_FEATURE_OPTIONS,
			required: false
		}));
		if (result.experimentFeatures.includes("vue-beta")) {
			const packageManagerOptions = getPackageManagerOptions(inferredPackageManager).map((pm) => ({
				value: pm,
				label: pm
			}));
			result.packageManager = await unwrapPrompt(ve({
				message: `${language.packageManagerSelection.message} ${(0, import_picocolors.dim)(language.packageManagerSelection.hint)}`,
				options: packageManagerOptions
			}));
		}
	}
	if (argv.bare) result.needsBareboneTemplates = true;
	else if (!isFeatureFlagsUsed) result.needsBareboneTemplates = await unwrapPrompt(ye({
		message: language.needsBareboneTemplates.message,
		initialValue: false
	}));
	const { features, experimentFeatures, needsBareboneTemplates } = result;
	const needsTypeScript = argv.ts || argv.typescript || features.includes("typescript");
	const needsJsx = argv.jsx || features.includes("jsx");
	const needsRouter = argv.router || argv["vue-router"] || features.includes("router");
	const needsPinia = argv.pinia || features.includes("pinia");
	const needsVitest = argv.vitest || argv.tests || features.includes("vitest");
	const needsEslint = argv.eslint || argv["eslint-with-prettier"] || features.includes("eslint");
	const needsPrettier = argv.prettier || argv["eslint-with-prettier"] || features.includes("prettier");
	const needsOxfmt = experimentFeatures.includes("oxfmt") || argv["oxfmt"];
	const needsViteBeta = experimentFeatures.includes("vite-beta") || argv["vite-beta"] || argv["rolldown-vite"];
	const needsVueBeta = experimentFeatures.includes("vue-beta") || argv["vue-beta"];
	const { e2eFramework } = result;
	const needsCypress = argv.cypress || argv.tests || e2eFramework === "cypress";
	const needsCypressCT = needsCypress && !needsVitest;
	const needsNightwatch = argv.nightwatch || e2eFramework === "nightwatch";
	const needsNightwatchCT = needsNightwatch && !needsVitest;
	const needsPlaywright = argv.playwright || e2eFramework === "playwright";
	const root = path$1.join(cwd, targetDir);
	if (fs.existsSync(root) && result.shouldOverwrite) emptyDir(root);
	else if (!fs.existsSync(root)) fs.mkdirSync(root);
	console.log(`\n${language.infos.scaffolding} ${root}...`);
	const pkg = {
		name: result.packageName,
		version: "0.0.0"
	};
	fs.writeFileSync(path$1.resolve(root, "package.json"), JSON.stringify(pkg, null, 2));
	const templateRoot = fileURLToPath(new URL("./template", import.meta.url));
	const callbacks = [];
	const render = function render(templateName) {
		renderTemplate(path$1.resolve(templateRoot, templateName), root, callbacks);
	};
	const replaceVite = () => {
		const content = fs.readFileSync(path$1.resolve(root, "package.json"), "utf-8");
		const json = JSON.parse(content);
		json.devDependencies.vite = "beta";
		fs.writeFileSync(path$1.resolve(root, "package.json"), JSON.stringify(json, null, 2));
	};
	render("base");
	if (needsJsx) render("config/jsx");
	if (needsRouter) render("config/router");
	if (needsPinia) render("config/pinia");
	if (needsVitest) render("config/vitest");
	if (needsCypress) render("config/cypress");
	if (needsCypressCT) render("config/cypress-ct");
	if (needsNightwatch) render("config/nightwatch");
	if (needsNightwatchCT) render("config/nightwatch-ct");
	if (needsPlaywright) render("config/playwright");
	if (needsTypeScript) {
		render("config/typescript");
		render("tsconfig/base");
		const rootTsConfig = {
			files: [],
			references: [{ path: "./tsconfig.node.json" }, { path: "./tsconfig.app.json" }]
		};
		if (needsCypress) {
			render("tsconfig/cypress");
			rootTsConfig.compilerOptions = { module: "NodeNext" };
		}
		if (needsCypressCT) {
			render("tsconfig/cypress-ct");
			rootTsConfig.references.push({ path: "./tsconfig.cypress-ct.json" });
		}
		if (needsPlaywright) render("tsconfig/playwright");
		if (needsVitest) {
			render("tsconfig/vitest");
			rootTsConfig.references.push({ path: "./tsconfig.vitest.json" });
		}
		if (needsNightwatch) {
			render("tsconfig/nightwatch");
			rootTsConfig.references.push({ path: "./nightwatch/tsconfig.json" });
		}
		if (needsNightwatchCT) render("tsconfig/nightwatch-ct");
		fs.writeFileSync(path$1.resolve(root, "tsconfig.json"), JSON.stringify(rootTsConfig, null, 2) + "\n", "utf-8");
	}
	if (needsEslint) {
		render("linting/base");
		if (needsTypeScript) render("linting/core/ts");
		else render("linting/core/js");
		if (needsCypress) render("linting/cypress");
		if (needsCypressCT) render("linting/cypress-ct");
		if (needsPlaywright) render("linting/playwright");
		if (needsVitest) render("linting/vitest");
		render("linting/oxlint");
		callbacks.push(async (dataStore) => {
			const oxlintrcPath = path$1.resolve(root, ".oxlintrc.json");
			dataStore[oxlintrcPath] = {
				needsTypeScript,
				needsVitest
			};
		});
		if (needsPrettier || needsOxfmt) render("linting/formatter");
	}
	if (needsOxfmt) render("formatting/oxfmt");
	else if (needsPrettier) render("formatting/prettier");
	if (needsViteBeta) replaceVite();
	render(`code/${(needsTypeScript ? "typescript-" : "") + (needsRouter ? "router" : "default")}`);
	if (needsPinia && needsRouter) render("entry/router-and-pinia");
	else if (needsPinia) render("entry/pinia");
	else if (needsRouter) render("entry/router");
	else render("entry/default");
	const dataStore = {};
	for (const cb of callbacks) await cb(dataStore);
	preOrderDirectoryTraverse(root, () => {}, (filepath) => {
		if (filepath.endsWith(".ejs")) {
			const template = fs.readFileSync(filepath, "utf-8");
			const dest = filepath.replace(/\.ejs$/, "");
			const content = import_ejs.render(template, dataStore[dest]);
			fs.writeFileSync(dest, content);
			fs.unlinkSync(filepath);
		}
	});
	if (needsBareboneTemplates) {
		trimBoilerplate(root);
		render("bare/base");
		if (needsTypeScript) render("bare/typescript");
		if (needsVitest) render("bare/vitest");
		if (needsCypressCT) render("bare/cypress-ct");
		if (needsNightwatchCT) render("bare/nightwatch-ct");
	}
	if (needsTypeScript) {
		preOrderDirectoryTraverse(root, () => {}, (filepath) => {
			if (filepath.endsWith(".js")) {
				const tsFilePath = filepath.replace(/\.js$/, ".ts");
				if (fs.existsSync(tsFilePath)) fs.unlinkSync(filepath);
				else fs.renameSync(filepath, tsFilePath);
			} else if (path$1.basename(filepath) === "jsconfig.json") fs.unlinkSync(filepath);
		});
		const indexHtmlPath = path$1.resolve(root, "index.html");
		const indexHtmlContent = fs.readFileSync(indexHtmlPath, "utf8");
		fs.writeFileSync(indexHtmlPath, indexHtmlContent.replace("src/main.js", "src/main.ts"));
	} else preOrderDirectoryTraverse(root, () => {}, (filepath) => {
		if (filepath.endsWith(".ts")) fs.unlinkSync(filepath);
	});
	if (needsBareboneTemplates) {
		removeCSSImport(root, needsTypeScript, needsCypressCT);
		if (needsRouter) emptyRouterConfig(root, needsTypeScript);
	}
	const packageManager = result.packageManager ?? inferredPackageManager;
	if (needsVueBeta) {
		const pkgPath = path$1.resolve(root, "package.json");
		const pkg = JSON.parse(fs.readFileSync(pkgPath, "utf-8"));
		applyVueBeta(root, packageManager, pkg);
		fs.writeFileSync(pkgPath, JSON.stringify(pkg, null, 2) + "\n");
	}
	fs.writeFileSync(path$1.resolve(root, "README.md"), generateReadme({
		projectName: result.projectName ?? result.packageName ?? defaultProjectName,
		packageManager,
		needsTypeScript,
		needsVitest,
		needsCypress,
		needsNightwatch,
		needsPlaywright,
		needsNightwatchCT,
		needsCypressCT,
		needsEslint
	}));
	let outroMessage = `${language.infos.done}\n\n`;
	if (root !== cwd) {
		const cdProjectName = path$1.relative(cwd, root);
		outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(`cd ${cdProjectName.includes(" ") ? `"${cdProjectName}"` : cdProjectName}`))}\n`;
	}
	outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(getCommand(packageManager, "install")))}\n`;
	if (needsPrettier || needsOxfmt) outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(getCommand(packageManager, "format")))}\n`;
	outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(getCommand(packageManager, "dev")))}\n`;
	if (!dotGitDirectoryState.hasDotGitDirectory) outroMessage += `
${(0, import_picocolors.dim)("|")} ${language.infos.optionalGitCommand}

   ${(0, import_picocolors.bold)((0, import_picocolors.green)("git init && git add -A && git commit -m \"initial commit\""))}`;
	Se(outroMessage);
}
init().catch((e) => {
	console.error(e);
	process.exit(1);
});

//#endregion
export {  };
